<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Creates Species Pools For Potential Habitats • SpeciesPoolR</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Creates Species Pools For Potential Habitats">
<meta name="description" content="What the package does (one paragraph).">
<meta property="og:description" content="What the package does (one paragraph).">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">SpeciesPoolR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/how_to_use.html">How to use the SpeciesPoolR package</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/derek-corcoran-barrios/SpeciesPoolR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><ul>
<li><a href="#id_1-speciespoolr">1 SpeciesPoolR</a></li>
<li>
<a href="#id_2-using-speciespoolr-manually">2 Using SpeciesPoolR Manually</a>
<ul>
<li><a href="#id_21-importing-and-downloading-species-presences">2.1 Importing and Downloading Species Presences</a></li>
<li><a href="#id_22-creating-spatial-buffers-and-habitat-filtering">2.2 Creating Spatial Buffers and Habitat Filtering</a></li>
<li><a href="#id_23-generating-summary-biodiversity-statistics">2.3 Generating summary biodiversity statistics</a></li>
</ul>
</li>
<li>
<a href="#id_3-running-the-speciespoolr-workflow">3 Running the SpeciesPoolR Workflow</a>
<ul>
<li><a href="#id_31-how-it-works">3.1 How It Works</a></li>
</ul>
</li>
<li><a href="#id_4-references">4 References</a></li>
</ul>
<!-- README.md is generated from README.Rmd. Please edit that file --><div class="section level1">
<div class="page-header"><h1 id="id_1-speciespoolr">1 SpeciesPoolR<a class="anchor" aria-label="anchor" href="#id_1-speciespoolr"></a>
</h1></div>
<!-- badges: start -->

<p>The goal of the SpeciesPoolR package is to generate potential species pools and their summary metrics in a spatial way. You can install the package directly from GitHub:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#install.packages("remotes")</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"derek-corcoran-barrios/SpeciesPoolR"</span><span class="op">)</span></span></code></pre></div>
<p>No you can load the package</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/derek-corcoran-barrios/SpeciesPoolR" class="external-link">SpeciesPoolR</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level1">
<h1 id="id_2-using-speciespoolr-manually">2 Using SpeciesPoolR Manually<a class="anchor" aria-label="anchor" href="#id_2-using-speciespoolr-manually"></a>
</h1>
<div class="section level2">
<h2 id="id_21-importing-and-downloading-species-presences">2.1 Importing and Downloading Species Presences<a class="anchor" aria-label="anchor" href="#id_21-importing-and-downloading-species-presences"></a>
</h2>
<div class="section level3">
<h3 id="id_211-step-1-reading-and-filtering-data">2.1.1 Step 1: Reading and Filtering Data<a class="anchor" aria-label="anchor" href="#id_211-step-1-reading-and-filtering-data"></a>
</h3>
<p>If you are going to use each of the functions of the SpeciesPoolR manually and sequentially, the first step would be to read in a species list from either a CSV or an XLSX file. You can use the get_data function for this. The function allows you to filter your data in a dplyr-like style:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/Species_List.csv"</span>, package<span class="op">=</span><span class="st">"SpeciesPoolR"</span><span class="op">)</span></span>
<span><span class="va">filtered_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_data.html">get_data</a></span><span class="op">(</span></span>
<span>   file <span class="op">=</span> <span class="va">f</span>,</span>
<span>   filter <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/substitute.html" class="external-link">quote</a></span><span class="op">(</span><span class="va">Kingdom</span> <span class="op">==</span> <span class="st">"Plantae"</span> <span class="op">&amp;</span> </span>
<span>                    <span class="va">Class</span> <span class="op">==</span> <span class="st">"Magnoliopsida"</span> <span class="op">&amp;</span> </span>
<span>                    <span class="va">Family</span> <span class="op">==</span> <span class="st">"Fabaceae"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>This will generate a dataset that can be used subsequently to count species presences and download species data as seen in table <a href="#tab:tablespecies">2.1</a></p>
<table class="table">
<colgroup>
<col width="13%">
<col width="8%">
<col width="14%">
<col width="14%">
<col width="8%">
<col width="9%">
<col width="10%">
<col width="21%">
</colgroup>
<thead><tr class="header">
<th align="left">redlist_2010</th>
<th align="left">Kingdom</th>
<th align="left">Phyllum</th>
<th align="left">Class</th>
<th align="left">Order</th>
<th align="left">Family</th>
<th align="left">Genus</th>
<th align="left">Species</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">NA</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Genista</td>
<td align="left">Genista tinctoria</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Trifolium</td>
<td align="left">Trifolium vesiculosum</td>
</tr>
<tr class="even">
<td align="left">LC</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sativa</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Lathyrus</td>
<td align="left">Lathyrus latifolius</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Anthyllis</td>
<td align="left">Anthyllis vulneraria</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Lathyrus</td>
<td align="left">Lathyrus japonicus</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia villosa</td>
</tr>
</tbody>
</table>
<p><span id="tab:tablespecies"></span>Table 2.1: Species that will be used to generate species pools</p>
</div>
<div class="section level3">
<h3 id="id_212-step-2-taxonomic-harmonization">2.1.2 Step 2: Taxonomic Harmonization<a class="anchor" aria-label="anchor" href="#id_212-step-2-taxonomic-harmonization"></a>
</h3>
<p>Next, you should perform taxonomic harmonization to ensure that the species names you use are recognized by the GBIF taxonomic backbone. This can be done using the Clean_Taxa function:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Clean_Species</span> <span class="op">&lt;-</span> <span class="fu">SpeciesPoolR</span><span class="fu">::</span><span class="fu"><a href="reference/Clean_Taxa.html">Clean_Taxa</a></span><span class="op">(</span><span class="va">filtered_data</span><span class="op">$</span><span class="va">Species</span><span class="op">)</span></span></code></pre></div>
<p>The resulting data frame, with harmonized species names, is shown in table <a href="#tab:cleantable">2.2</a></p>
<table style="width:100%;" class="table">
<colgroup>
<col width="12%">
<col width="12%">
<col width="6%">
<col width="12%">
<col width="4%">
<col width="7%">
<col width="8%">
<col width="4%">
<col width="5%">
<col width="6%">
<col width="12%">
<col width="4%">
</colgroup>
<thead><tr class="header">
<th align="left">Taxa</th>
<th align="left">matched_name2</th>
<th align="right">confidence</th>
<th align="left">canonicalName</th>
<th align="left">kingdom</th>
<th align="left">phylum</th>
<th align="left">class</th>
<th align="left">order</th>
<th align="left">family</th>
<th align="left">genus</th>
<th align="left">species</th>
<th align="left">rank</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Vicia sepium</td>
<td align="left">Vicia sepium</td>
<td align="right">99</td>
<td align="left">Vicia sepium</td>
<td align="left">Plantae</td>
<td align="left">Tracheophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sepium</td>
<td align="left">SPECIES</td>
</tr>
<tr class="even">
<td align="left">Genista tinctoria</td>
<td align="left">Genista tinctoria</td>
<td align="right">99</td>
<td align="left">Genista tinctoria</td>
<td align="left">Plantae</td>
<td align="left">Tracheophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Genista</td>
<td align="left">Genista tinctoria</td>
<td align="left">SPECIES</td>
</tr>
<tr class="odd">
<td align="left">Trifolium vesiculosum</td>
<td align="left">Trifolium vesiculosum</td>
<td align="right">99</td>
<td align="left">Trifolium vesiculosum</td>
<td align="left">Plantae</td>
<td align="left">Tracheophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Trifolium</td>
<td align="left">Trifolium vesiculosum</td>
<td align="left">SPECIES</td>
</tr>
<tr class="even">
<td align="left">Vicia sativa</td>
<td align="left">Vicia sativa</td>
<td align="right">97</td>
<td align="left">Vicia sativa</td>
<td align="left">Plantae</td>
<td align="left">Tracheophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sativa</td>
<td align="left">SPECIES</td>
</tr>
<tr class="odd">
<td align="left">Lathyrus latifolius</td>
<td align="left">Lathyrus latifolius</td>
<td align="right">98</td>
<td align="left">Lathyrus latifolius</td>
<td align="left">Plantae</td>
<td align="left">Tracheophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Lathyrus</td>
<td align="left">Lathyrus latifolius</td>
<td align="left">SPECIES</td>
</tr>
<tr class="even">
<td align="left">Anthyllis vulneraria</td>
<td align="left">Anthyllis vulneraria</td>
<td align="right">97</td>
<td align="left">Anthyllis vulneraria</td>
<td align="left">Plantae</td>
<td align="left">Tracheophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Anthyllis</td>
<td align="left">Anthyllis vulneraria</td>
<td align="left">SPECIES</td>
</tr>
<tr class="odd">
<td align="left">Lathyrus japonicus</td>
<td align="left">Lathyrus japonicus</td>
<td align="right">99</td>
<td align="left">Lathyrus japonicus</td>
<td align="left">Plantae</td>
<td align="left">Tracheophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Lathyrus</td>
<td align="left">Lathyrus japonicus</td>
<td align="left">SPECIES</td>
</tr>
<tr class="even">
<td align="left">Vicia villosa</td>
<td align="left">Vicia villosa</td>
<td align="right">97</td>
<td align="left">Vicia villosa</td>
<td align="left">Plantae</td>
<td align="left">Tracheophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia villosa</td>
<td align="left">SPECIES</td>
</tr>
</tbody>
</table>
<p><span id="tab:cleantable"></span>Table 2.2: Taxonomicallty harmonized dataset</p>
</div>
<div class="section level3">
<h3 id="id_213-step-3-counting-species-presences">2.1.3 Step 3: Counting Species Presences<a class="anchor" aria-label="anchor" href="#id_213-step-3-counting-species-presences"></a>
</h3>
<p>After harmonizing the species names, it’s important to obtain the number of occurrences of each species in your study area, especially if you plan to calculate rarity. You can do this using the <code>count_presences</code> function. This function allows you to filter occurrences by country or by a shapefile. Below is an example for Denmark:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Assuming Clean_Species is your data frame</span></span>
<span><span class="va">Count_DK</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/count_presences.html">count_presences</a></span><span class="op">(</span><span class="va">Clean_Species</span>, country <span class="op">=</span> <span class="st">"DK"</span><span class="op">)</span></span></code></pre></div>
<p>The resulting data frame of species presences in Denmark is shown in table <a href="#tab:tableCountDenmark">2.3</a></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">Count_DK</span>, caption <span class="op">=</span> <span class="st">"Counts of presences for the different species within Denmark"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">family</th>
<th align="left">genus</th>
<th align="left">species</th>
<th align="right">N</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sepium</td>
<td align="right">2897</td>
</tr>
<tr class="even">
<td align="left">Fabaceae</td>
<td align="left">Genista</td>
<td align="left">Genista tinctoria</td>
<td align="right">988</td>
</tr>
<tr class="odd">
<td align="left">Fabaceae</td>
<td align="left">Trifolium</td>
<td align="left">Trifolium vesiculosum</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sativa</td>
<td align="right">17379</td>
</tr>
<tr class="odd">
<td align="left">Fabaceae</td>
<td align="left">Lathyrus</td>
<td align="left">Lathyrus latifolius</td>
<td align="right">684</td>
</tr>
<tr class="even">
<td align="left">Fabaceae</td>
<td align="left">Anthyllis</td>
<td align="left">Anthyllis vulneraria</td>
<td align="right">8876</td>
</tr>
<tr class="odd">
<td align="left">Fabaceae</td>
<td align="left">Lathyrus</td>
<td align="left">Lathyrus japonicus</td>
<td align="right">3904</td>
</tr>
<tr class="even">
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia villosa</td>
<td align="right">243</td>
</tr>
</tbody>
</table>
<p><span id="tab:tableCountDenmark"></span>Table 2.3: Counts of presences for the different species within Denmark</p>
<p>Alternatively, you can filter by a specific region using a shapefile. For example, to count species presences within Aarhus commune:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">shp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/Aarhus.shp"</span>, package<span class="op">=</span><span class="st">"SpeciesPoolR"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Count_Aarhus</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/count_presences.html">count_presences</a></span><span class="op">(</span><span class="va">Clean_Species</span>, shapefile <span class="op">=</span> <span class="va">shp</span><span class="op">)</span></span></code></pre></div>
<p>The resulting data.frame for Aarhus commune is shown int table <a href="#tab:tableCountAarhus">2.4</a></p>
<table class="table">
<thead><tr class="header">
<th align="left">family</th>
<th align="left">genus</th>
<th align="left">species</th>
<th align="right">N</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sepium</td>
<td align="right">283</td>
</tr>
<tr class="even">
<td align="left">Fabaceae</td>
<td align="left">Genista</td>
<td align="left">Genista tinctoria</td>
<td align="right">27</td>
</tr>
<tr class="odd">
<td align="left">Fabaceae</td>
<td align="left">Trifolium</td>
<td align="left">Trifolium vesiculosum</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sativa</td>
<td align="right">467</td>
</tr>
<tr class="odd">
<td align="left">Fabaceae</td>
<td align="left">Lathyrus</td>
<td align="left">Lathyrus latifolius</td>
<td align="right">41</td>
</tr>
<tr class="even">
<td align="left">Fabaceae</td>
<td align="left">Anthyllis</td>
<td align="left">Anthyllis vulneraria</td>
<td align="right">153</td>
</tr>
<tr class="odd">
<td align="left">Fabaceae</td>
<td align="left">Lathyrus</td>
<td align="left">Lathyrus japonicus</td>
<td align="right">39</td>
</tr>
<tr class="even">
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia villosa</td>
<td align="right">10</td>
</tr>
</tbody>
</table>
<p><span id="tab:tableCountAarhus"></span>Table 2.4: Counts of presences for the different species within Aarhus commune</p>
<p>Now it is recommended to eliminate species that have no occurrences in the area, this is done automatically in the workflow version:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com" class="external-link">data.table</a></span><span class="op">)</span></span>
<span><span class="va">Count_Aarhus</span> <span class="op">&lt;-</span> <span class="va">Count_Aarhus</span><span class="op">[</span><span class="va">N</span> <span class="op">&gt;</span> <span class="fl">0</span>,<span class="op">]</span></span></code></pre></div>
<p>So that then we can retrieve the species presences using the function <code><a href="reference/get_presences.html">SpeciesPoolR::get_presences</a></code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Presences</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_presences.html">get_presences</a></span><span class="op">(</span>species <span class="op">=</span> <span class="va">Count_Aarhus</span><span class="op">$</span><span class="va">species</span>, shapefile <span class="op">=</span> <span class="va">shp</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Geometry created: POLYGON ((10.401438 56.302419, 10.048024 56.355225, 9.886316 56.019928, 10.239729 55.966657, 10.401438 56.302419))"</span></span></code></pre></div>
<p>there we end up with 1070 presences for our 7 species.</p>
</div>
</div>
<div class="section level2">
<h2 id="id_22-creating-spatial-buffers-and-habitat-filtering">2.2 Creating Spatial Buffers and Habitat Filtering<a class="anchor" aria-label="anchor" href="#id_22-creating-spatial-buffers-and-habitat-filtering"></a>
</h2>
<div class="section level3">
<h3 id="id_221-step-1-creating-buffers-around-species-presences">2.2.1 Step 1 Creating Buffers Around Species Presences<a class="anchor" aria-label="anchor" href="#id_221-step-1-creating-buffers-around-species-presences"></a>
</h3>
<p>Once you have obtained the species presences within your area of interest, the next step is to create spatial buffers around these occurrences. These buffers simulate the potential dispersal range of each species. By doing so, you can assess the possible areas where each species could establish itself, considering a given dispersal distance.</p>
<p>To create these buffers, you’ll need a raster file that serves as a template for rasterizing the buffers. You’ll also specify the distance (in meters) that represents the dispersal range of the species.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Raster</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/LU_Aarhus.tif"</span>, package<span class="op">=</span><span class="st">"SpeciesPoolR"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">buffer500</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/make_buffer_rasterized.html">make_buffer_rasterized</a></span><span class="op">(</span><span class="va">Presences</span>, file <span class="op">=</span> <span class="va">Raster</span>, dist <span class="op">=</span> <span class="fl">500</span><span class="op">)</span></span></code></pre></div>
<p>In the example above, the <code>make_buffer_rasterized</code> function creates a 500-meter buffer around each occurrence point of the species in the Presences dataset. The function uses the provided raster file as a template to rasterize the buffers.</p>
<p>The resulting data frame, <code>buffer500</code>, contains information on which raster cells are covered by the buffer for each species. The first 10 observations of this data frame are shown in Table <a href="#tab:showbuffer500">2.5</a>.</p>
<table class="table">
<thead><tr class="header">
<th align="right">cell</th>
<th align="left">species</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">26</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="even">
<td align="right">27</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="odd">
<td align="right">28</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="even">
<td align="right">29</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="odd">
<td align="right">30</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="even">
<td align="right">161</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="odd">
<td align="right">162</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="even">
<td align="right">163</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="odd">
<td align="right">164</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="even">
<td align="right">165</td>
<td align="left">Vicia sepium</td>
</tr>
</tbody>
</table>
<p><span id="tab:showbuffer500"></span>Table 2.5: Raster cells within the 500-meter buffer of each species</p>
<p>This table provides a detailed view of how the buffer overlaps with the raster cells, listing each cell and the corresponding species present within that buffer.</p>
</div>
<div class="section level3">
<h3 id="id_222-step-2-habitat-filtering">2.2.2 Step 2: Habitat Filtering<a class="anchor" aria-label="anchor" href="#id_222-step-2-habitat-filtering"></a>
</h3>
<p>After creating the buffers, you may want to filter the areas further based on habitat suitability. For instance, you might only be interested in specific land-use types or habitats where the species is likely to thrive. This step typically involves using the raster data to mask or subset the buffer areas according to the desired habitat criteria.</p>
</div>
</div>
<div class="section level2">
<h2 id="id_23-generating-summary-biodiversity-statistics">2.3 Generating summary biodiversity statistics<a class="anchor" aria-label="anchor" href="#id_23-generating-summary-biodiversity-statistics"></a>
</h2>
<div class="section level3">
<h3 id="id_231-step-1-generating-phylogenetic-diversity-metrics">2.3.1 Step 1 Generating Phylogenetic diversity metrics<a class="anchor" aria-label="anchor" href="#id_231-step-1-generating-phylogenetic-diversity-metrics"></a>
</h3>
<p>In order to generate Phylogenetic Diversity measures, the first step is to generate a phylogenetic tree with the species we have, for that we will use the V.Phylomaker package function <code>phylo.maker</code>based on the megaphylogeny of vascular plants (Jin and Qian 2019; Zanne et al. 2014), this means that we can only use this functions in species pools of plants.</p>
<p>In this case we use the <code>generate_tree</code> from SpeciesPoolR to do so:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/generate_tree.html">generate_tree</a></span><span class="op">(</span><span class="va">Count_Aarhus</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "All species in sp.list are present on tree."</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level1">
<h1 id="id_3-running-the-speciespoolr-workflow">3 Running the SpeciesPoolR Workflow<a class="anchor" aria-label="anchor" href="#id_3-running-the-speciespoolr-workflow"></a>
</h1>
<p>If you prefer to automate the process and run the <code>SpeciesPoolR</code> workflow as a pipeline, you can use the <code>run_workflow</code> function. This function sets up a <code>targets</code> workflow that sequentially executes the steps for cleaning species data, counting species presences, and performing spatial analysis. This approach is especially useful for larger datasets or when you want to ensure reproducibility.</p>
<p>To run the workflow, you can use the following code. We’ll use the same species filter as before, focusing on the <code>Plantae</code> kingdom, <code>Magnoliopsida</code> class, and <code>Fabaceae</code> family. Additionally, we’ll focus on the Aarhus commune using a shapefile.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">shp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/Aarhus.shp"</span>, package <span class="op">=</span> <span class="st">"SpeciesPoolR"</span><span class="op">)</span></span>
<span><span class="va">Raster</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/LU_Aarhus.tif"</span>, package<span class="op">=</span><span class="st">"SpeciesPoolR"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="reference/run_workflow.html">run_workflow</a></span><span class="op">(</span></span>
<span>  file_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/Species_List.csv"</span>, package <span class="op">=</span> <span class="st">"SpeciesPoolR"</span><span class="op">)</span>,</span>
<span>  filter <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/substitute.html" class="external-link">quote</a></span><span class="op">(</span><span class="va">Kingdom</span> <span class="op">==</span> <span class="st">"Plantae"</span> <span class="op">&amp;</span> <span class="va">Class</span> <span class="op">==</span> <span class="st">"Magnoliopsida"</span> <span class="op">&amp;</span> <span class="va">Family</span> <span class="op">==</span> <span class="st">"Fabaceae"</span><span class="op">)</span>,</span>
<span>  shapefile <span class="op">=</span> <span class="va">shp</span>,</span>
<span>  dist <span class="op">=</span> <span class="fl">500</span>,</span>
<span>  rastertemp <span class="op">=</span> <span class="va">Raster</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; ▶ dispatched target Raster</span></span>
<span><span class="co">#&gt; ▶ dispatched target shp</span></span>
<span><span class="co">#&gt; ● completed target Raster [4.523 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched target file</span></span>
<span><span class="co">#&gt; ● completed target shp [0 seconds]</span></span>
<span><span class="co">#&gt; ● completed target file [0 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched target data</span></span>
<span><span class="co">#&gt; ● completed target data [0.357 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched target Clean</span></span>
<span><span class="co">#&gt; ● completed target Clean [1.185 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Count_Presences_33538e94b3809372</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Count_Presences_52d72a5ad405e933</span></span>
<span><span class="co">#&gt; ● completed branch Count_Presences_33538e94b3809372 [0.149 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Count_Presences_e70f77d9439a4770</span></span>
<span><span class="co">#&gt; ● completed branch Count_Presences_52d72a5ad405e933 [0.057 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Count_Presences_dea4ef8633a449a1</span></span>
<span><span class="co">#&gt; ● completed branch Count_Presences_e70f77d9439a4770 [0.029 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Count_Presences_69210fc440d13855</span></span>
<span><span class="co">#&gt; ● completed branch Count_Presences_dea4ef8633a449a1 [0.031 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Count_Presences_a61be030e01ebaf5</span></span>
<span><span class="co">#&gt; ● completed branch Count_Presences_69210fc440d13855 [0.026 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Count_Presences_974105e269324d3e</span></span>
<span><span class="co">#&gt; ● completed branch Count_Presences_a61be030e01ebaf5 [0.028 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Count_Presences_37d1f8d5f74d852c</span></span>
<span><span class="co">#&gt; ● completed branch Count_Presences_974105e269324d3e [0.036 seconds]</span></span>
<span><span class="co">#&gt; ● completed branch Count_Presences_37d1f8d5f74d852c [0.035 seconds]</span></span>
<span><span class="co">#&gt; ● completed pattern Count_Presences</span></span>
<span><span class="co">#&gt; ▶ dispatched target More_than_zero</span></span>
<span><span class="co">#&gt; ● completed target More_than_zero [0.001 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Presences_c112b37cd15959d6</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Presences_af64bac105a08467</span></span>
<span><span class="co">#&gt; ● completed branch Presences_c112b37cd15959d6 [0.684 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch buffer_626a53b08dfe709d</span></span>
<span><span class="co">#&gt; ● completed branch Presences_af64bac105a08467 [0.313 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch buffer_0e19b8cb545404d2</span></span>
<span><span class="co">#&gt; ● completed branch buffer_626a53b08dfe709d [0.108 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Presences_daf8d6353bc80f0c</span></span>
<span><span class="co">#&gt; ● completed branch buffer_0e19b8cb545404d2 [0.236 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Presences_310adeccf6b44725</span></span>
<span><span class="co">#&gt; ● completed branch Presences_daf8d6353bc80f0c [0.666 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch buffer_edb09c8ec5c9a988</span></span>
<span><span class="co">#&gt; ● completed branch Presences_310adeccf6b44725 [0.367 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch buffer_b226446ac3154351</span></span>
<span><span class="co">#&gt; ● completed branch buffer_edb09c8ec5c9a988 [0.043 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Presences_e65f4227e8299cc4</span></span>
<span><span class="co">#&gt; ● completed branch buffer_b226446ac3154351 [0.028 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Presences_d4b9dc68293bd5b2</span></span>
<span><span class="co">#&gt; ● completed branch Presences_e65f4227e8299cc4 [0.356 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch buffer_0a8436ee3d4f2644</span></span>
<span><span class="co">#&gt; ● completed branch Presences_d4b9dc68293bd5b2 [0.333 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch buffer_cae8301e59fc4e01</span></span>
<span><span class="co">#&gt; ● completed branch buffer_0a8436ee3d4f2644 [0.035 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Presences_88937156c1302a12</span></span>
<span><span class="co">#&gt; ● completed branch buffer_cae8301e59fc4e01 [0.028 seconds]</span></span>
<span><span class="co">#&gt; ● completed branch Presences_88937156c1302a12 [0.294 seconds]</span></span>
<span><span class="co">#&gt; ● completed pattern Presences</span></span>
<span><span class="co">#&gt; ▶ dispatched branch buffer_a0190cbfdf5f6f1f</span></span>
<span><span class="co">#&gt; ● completed branch buffer_a0190cbfdf5f6f1f [0.032 seconds]</span></span>
<span><span class="co">#&gt; ● completed pattern buffer</span></span>
<span><span class="co">#&gt; ▶ ended pipeline [12.674 seconds]</span></span></code></pre></div>
<p><img src="reference/figures/README-run_workflow-1.png" width="100%"></p>
<div class="section level2">
<h2 id="id_31-how-it-works">3.1 How It Works<a class="anchor" aria-label="anchor" href="#id_31-how-it-works"></a>
</h2>
<p>The run_workflow function creates a pipeline that:</p>
<p>1- Reads the data from the specified file path. 2- Filters the data using the provided filter expression. 3- Cleans the species names to match the GBIF taxonomic backbone. 4- Counts the species presences within the specified geographic area (in this case, Aarhus). 5- Generates a buffer around the species presences within the specified distance, for a template raster. 6- Generates a visual representation of the workflow (if plot = TRUE).</p>
<p>You can monitor the progress of the workflow and visualize the dependencies between steps using targets::tar_visnetwork(). The result will be similar to running the steps manually but with the added benefits of parallel execution and reproducibility.</p>
<p>This automated approach allows you to streamline your analysis and ensures that all steps are consistently applied to your data. It also makes it easier to rerun the workflow with different parameters or datasets.</p>
</div>
</div>
<div class="section level1">
<h1 id="id_4-references">4 References<a class="anchor" aria-label="anchor" href="#id_4-references"></a>
</h1>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Jin2019" class="csl-entry">
Jin, Yi, and Hong Qian. 2019. “V.PhyloMaker: An r Package That Can Generate Very Large Phylogenies for Vascular Plants.” <em>Ecography</em> 42: 1353–59.
</div>
<div id="ref-Zanne2014" class="csl-entry">
Zanne, Amy E., David C. Tank, William K. Cornwell, Jonathan M. Eastman, Stephen A. Smith, Richard G. FitzJohn, Daniel J. McGlinn, et al. 2014. “Three Keys to the Radiation of Angiosperms into Freezing Environments.” <em>American Journal of Botany</em> 506: 89–92.
</div>
</div>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/derek-corcoran-barrios/SpeciesPoolR/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/derek-corcoran-barrios/SpeciesPoolR/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing SpeciesPoolR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Derek Corcoran <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-0248-6230" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/derek-corcoran-barrios/SpeciesPoolR/actions/workflows/R-CMD-check.yaml" class="external-link"><img src="https://github.com/derek-corcoran-barrios/SpeciesPoolR/actions/workflows/R-CMD-check.yaml/badge.svg" alt="R-CMD-check"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Derek Corcoran.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
