<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Creates Species Pools For Potential Habitats • SpeciesPoolR</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Creates Species Pools For Potential Habitats">
<meta name="description" content="What the package does (one paragraph).">
<meta property="og:description" content="What the package does (one paragraph).">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">SpeciesPoolR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/how_to_use.html">How to use the SpeciesPoolR package</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/derek-corcoran-barrios/SpeciesPoolR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><ul>
<li><a href="#id_1-speciespoolr">1 SpeciesPoolR</a></li>
<li>
<a href="#id_2-motivation-for-the-pacakge">2 Motivation for the pacakge</a>
<ul>
<li><a href="#id_21-rare-species-are-common-and-important">2.1 Rare species are common and important</a></li>
</ul>
</li>
<li>
<a href="#id_3-required-data-files">3 Required Data Files</a>
<ul>
<li><a href="#id_31-species-list-file">3.1 Species List File</a></li>
<li><a href="#id_32-shapefile">3.2 Shapefile</a></li>
<li><a href="#id_33-raster-template-file">3.3 Raster Template File</a></li>
<li><a href="#id_34-land-use-raster-file">3.4 Land-Use Raster File</a></li>
<li><a href="#id_35-land-use-suitability-raster-file">3.5 Land-Use Suitability Raster File</a></li>
</ul>
</li>
<li>
<a href="#id_4-using-speciespoolr-manually">4 Using SpeciesPoolR Manually</a>
<ul>
<li><a href="#id_41-importing-and-downloading-species-presences">4.1 Importing and Downloading Species Presences</a></li>
<li><a href="#id_42-creating-spatial-buffers-and-habitat-filtering">4.2 Creating Spatial Buffers and Habitat Filtering</a></li>
<li><a href="#id_43-generating-summary-biodiversity-statistics">4.3 Generating summary biodiversity statistics</a></li>
</ul>
</li>
<li>
<a href="#id_5-running-the-speciespoolr-workflow">5 Running the SpeciesPoolR Workflow</a>
<ul>
<li><a href="#id_51-how-it-works">5.1 How It Works</a></li>
</ul>
</li>
<li><a href="#id_6-references">6 References</a></li>
</ul>
<!-- README.md is generated from README.Rmd. Please edit that file --><div class="section level1">
<div class="page-header"><h1 id="id_1-speciespoolr">1 SpeciesPoolR<a class="anchor" aria-label="anchor" href="#id_1-speciespoolr"></a>
</h1></div>
<!-- badges: start -->

<!-- badges: end -->
<p>The goal of the SpeciesPoolR package is to generate potential species pools and their summary metrics in a spatial way. You can install the package directly from GitHub:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#install.packages("remotes")</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"derek-corcoran-barrios/SpeciesPoolR"</span><span class="op">)</span></span></code></pre></div>
<p>No you can load the package</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/derek-corcoran-barrios/SpeciesPoolR" class="external-link">SpeciesPoolR</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level1">
<h1 id="id_2-motivation-for-the-pacakge">2 Motivation for the pacakge<a class="anchor" aria-label="anchor" href="#id_2-motivation-for-the-pacakge"></a>
</h1>
<div class="section level2">
<h2 id="id_21-rare-species-are-common-and-important">2.1 Rare species are common and important<a class="anchor" aria-label="anchor" href="#id_21-rare-species-are-common-and-important"></a>
</h2>
<p>In ecological research, the debate on whether rare species outnumber common species within communities is pivotal for understanding biodiversity and guiding conservation efforts. Numerous studies have shown that rare species typically dominate large ecological assemblages, although common species often exert a more substantial influence on overall species richness patterns (Magurran and Henderson 2003; Bregović, Fišer, and Zagmajster 2019; Schalkwyk, Pryke, and Samways 2019). This complexity underscores the need for innovative approaches in studying biodiversity, particularly since rare species are challenging to model using traditional Species Distribution Models (SDMs) due to their low occurrence rates (Boyd et al. 2022).</p>
<p>Given the limitations of SDMs in capturing the dynamics of rare species, it is essential to develop alternative methods for integrating these species into biodiversity assessments and conservation planning. Although rare species contribute uniquely to functional diversity and ecosystem stability, especially in specific habitats (Chapman, Tunnicliffe, and Bates 2018; Säterberg et al. 2019), their elusiveness in ecological models presents a significant challenge. The question of the minimum number of presence records required for reliable SDMs is crucial. Research has shown that while as few as 10-15 presence observations can produce nonrandom models for some species (Støa et al. 2019), others require higher thresholds—ranging from 14 to 25 records depending on the species’ prevalence and geographic range (Proosdij et al. 2016; Sampaio and Cavalcante 2023). These findings suggest that even sparse datasets can be useful, but the threshold varies significantly depending on species traits and habitat characteristics. Therefore, researchers must explore novel analytical frameworks and conservation strategies that better accommodate the ecological importance of rare species, thereby enhancing our ability to manage and preserve biodiversity effectively (Reddin, Bothwell, and Lennon 2015).</p>
<p>In highly degraded habitats, such as Denmark, where over 60% of the land is dominated by agriculture and less than 10% remains as natural habitat, traditional SDMs may face further limitations. The scarcity of natural habitats means that presence records are often skewed towards human-modified landscapes, complicating the modeling of species’ ecological preferences. In such contexts, where the majority of occurrences may not reflect the species’ natural behaviors or habitat use, relying on complex SDMs could lead to misleading predictions. Instead, simpler algorithms that incorporate basic dispersal mechanisms and habitat filtering might be more effective. By reducing assumptions about habitat preferences, these methods can provide a more realistic framework for conservation planning, particularly when dealing with the restoration of agricultural lands into natural habitats.</p>
<p>For rare species, and indeed for many others, this approach may offer a more practical solution in scenarios where detailed ecological data is sparse or unreliable. Studies have suggested that in such landscapes, simplistic models that prioritize dispersal and broad habitat suitability over intricate ecological niches can better capture species’ potential distributions and their responses to environmental changes (Guisan et al. 2006; Thuiller et al. 2005), an example to this approach would be range bagging (Drake 2015). This pragmatic approach is especially pertinent when planning conservation actions in areas where habitat degradation has left little intact nature, and it ensures that even under data constraints, effective biodiversity management can still be pursued.</p>
</div>
</div>
<div class="section level1">
<h1 id="id_3-required-data-files">3 Required Data Files<a class="anchor" aria-label="anchor" href="#id_3-required-data-files"></a>
</h1>
<p>To effectively execute the <code>SpeciesPoolR</code> workflow, a set of essential data files must be provided. These files contain the necessary spatial and taxonomic information that underpin the various analytical steps in the package. Below, we detail each required file and its role within the workflow.</p>
<div class="section level2">
<h2 id="id_31-species-list-file">3.1 Species List File<a class="anchor" aria-label="anchor" href="#id_31-species-list-file"></a>
</h2>
<ul>
<li><p><strong>File Type</strong>: CSV or Excel file</p></li>
<li><p><strong>Description</strong>: The species list file serves as the foundational dataset, comprising the species of interest for your analysis. At a minimum, this file must include a column for the scientific names of species (<code>Species</code>). Additional taxonomic columns, such as <code>Kingdom</code>, <code>Class</code>, and <code>Family</code>, may also be included to facilitate filtering and subgroup analyses.</p></li>
</ul>
<p>An example of this file is provided within the package and can be accessed using the following code:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exampleSpecies</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/Species_List.csv"</span>, package<span class="op">=</span><span class="st">"SpeciesPoolR"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">exampleSpecies</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "/home/au687614/R/x86_64-pc-linux-gnu-library/4.4/SpeciesPoolR/ex/Species_List.csv"</span></span></code></pre></div>
<p>This dataset is further discussed in the section on <a href="#id_411-step-1-reading-and-filtering-data">Reading and Filtering Data</a>, with a filtered subset displayed in Table <a href="#tab:tablespecies">4.1</a>.</p>
</div>
<div class="section level2">
<h2 id="id_32-shapefile">3.2 Shapefile<a class="anchor" aria-label="anchor" href="#id_32-shapefile"></a>
</h2>
<ul>
<li><p><strong>File Type</strong>: Shapefile (.shp)</p></li>
<li><p><strong>Description</strong>: The shapefile delineates the geographic area of interest, which can range from a broad region, such as a country, to a more specific locality, such as a nature reserve. This file is utilized to spatially constrain species occurrences, ensuring that only those within the defined boundaries are included in the analysis.</p></li>
</ul>
<p>If a shapefile is unavailable, a two-letter country code (e.g., “DK” for Denmark) may be provided as an alternative to specify the area of interest.</p>
<p>An example shapefile is included in the package and can be accessed as follows:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">shp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/Aarhus.shp"</span>, package<span class="op">=</span><span class="st">"SpeciesPoolR"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">shp</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "/home/au687614/R/x86_64-pc-linux-gnu-library/4.4/SpeciesPoolR/ex/Aarhus.shp"</span></span></code></pre></div>
<p>The shapefile’s application is illustrated in the section on <a href="#id_413-step-3-counting-species-presences">Counting Species Presences</a>, where it is used to delineate the boundaries of Aarhus commune, as shown in Figure <a href="#fig:plotshapefile">3.1</a>.</p>
<div class="figure">
<p><img src="reference/figures/README-plotshapefile-1.png" alt="Outline of the comune of Aarhus" width="100%"></p>
<p class="caption">
</p>
<p><span id="fig:plotshapefile"></span>Figure 3.1: Outline of the comune of Aarhus</p>

</div>
</div>
<div class="section level2">
<h2 id="id_33-raster-template-file">3.3 Raster Template File<a class="anchor" aria-label="anchor" href="#id_33-raster-template-file"></a>
</h2>
<ul>
<li><p><strong>File Type</strong>: Raster file (e.g., .tif)</p></li>
<li><p><strong>Description</strong>: The raster template file is employed as a spatial reference for rasterizing species presence buffers. It must cover the entire area of interest and possess a resolution appropriate for the intended analysis. This template ensures consistent spatial alignment across all raster-based operations.</p></li>
</ul>
<p>You can explore an example of this file using the following code:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">template</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/LU_Aarhus.tif"</span>, package<span class="op">=</span><span class="st">"SpeciesPoolR"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">template</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "/home/au687614/R/x86_64-pc-linux-gnu-library/4.4/SpeciesPoolR/ex/LU_Aarhus.tif"</span></span></code></pre></div>
<p>The raster template’s role in buffer creation is further explained in the section on <a href="#id_421-step-1-creating-buffers-around-species-presences">Creating Buffers Around Species Presences</a>, with an example shown in Figure <a href="#fig:plottemplate">3.2</a>.</p>
<div class="figure">
<p><img src="reference/figures/README-plottemplate-1.png" alt="Raster of the Aarhus comune, the package will use Non NA cells as part of the template" width="100%"></p>
<p class="caption">
</p>
<p><span id="fig:plottemplate"></span>Figure 3.2: Raster of the Aarhus comune, the package will use Non NA cells as part of the template</p>

</div>
</div>
<div class="section level2">
<h2 id="id_34-land-use-raster-file">3.4 Land-Use Raster File<a class="anchor" aria-label="anchor" href="#id_34-land-use-raster-file"></a>
</h2>
<ul>
<li><p><strong>File Type</strong>: Raster file (e.g., .tif)</p></li>
<li><p><strong>Description</strong>: This file contains land-use classifications for the study area, where each raster cell is assigned to a specific land-use category (e.g., forest, wetland, urban). This data is crucial for modeling habitat suitability, enabling the filtering of species occurrences based on the prevalent land uses within their potential habitats.</p></li>
</ul>
<p>An example file is provided in the package:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">LU</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/LU_Aarhus.tif"</span>, package<span class="op">=</span><span class="st">"SpeciesPoolR"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">LU</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "/home/au687614/R/x86_64-pc-linux-gnu-library/4.4/SpeciesPoolR/ex/LU_Aarhus.tif"</span></span></code></pre></div>
<p>The land-use raster is identical to the template shown in Figure <a href="#fig:plottemplate">3.2</a>.</p>
</div>
<div class="section level2">
<h2 id="id_35-land-use-suitability-raster-file">3.5 Land-Use Suitability Raster File<a class="anchor" aria-label="anchor" href="#id_35-land-use-suitability-raster-file"></a>
</h2>
<ul>
<li><p><strong>File Type</strong>: Raster file (e.g., .tif)</p></li>
<li><p><strong>Description</strong>: This file comprises binary suitability values for various land-use types within the study area, indicating whether each land-use type is suitable (value = 1) or unsuitable (value = 0) for the habitat of interest. The data is subsequently transformed into a long-format table, which is integral to the habitat filtering and species distribution modeling processes.</p></li>
</ul>
<p>An example raster file is available in the package, and its application is discussed in the section on <a href="#id_4221-preparing-land-use-data">Preparing Land-Use Data</a>. A visualization of this file is presented in Figure <a href="#fig:plotexampleLU">3.3</a>.</p>
<div class="figure">
<p><img src="reference/figures/README-plotexampleLU-1.png" alt="Landuse suitability for 8 different landuses in the aarhus commune" width="100%"></p>
<p class="caption">
</p>
<p><span id="fig:plotexampleLU"></span>Figure 3.3: Landuse suitability for 8 different landuses in the aarhus commune</p>

</div>
</div>
</div>
<div class="section level1">
<h1 id="id_4-using-speciespoolr-manually">4 Using SpeciesPoolR Manually<a class="anchor" aria-label="anchor" href="#id_4-using-speciespoolr-manually"></a>
</h1>
<div class="section level2">
<h2 id="id_41-importing-and-downloading-species-presences">4.1 Importing and Downloading Species Presences<a class="anchor" aria-label="anchor" href="#id_41-importing-and-downloading-species-presences"></a>
</h2>
<div class="section level3">
<h3 id="id_411-step-1-reading-and-filtering-data">4.1.1 Step 1: Reading and Filtering Data<a class="anchor" aria-label="anchor" href="#id_411-step-1-reading-and-filtering-data"></a>
</h3>
<p>If you are going to use each of the functions of the SpeciesPoolR manually and sequentially, the first step would be to read in a species list from either a CSV or an XLSX file. You can use the get_data function for this. The function allows you to filter your data in a dplyr-like style:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/Species_List.csv"</span>, package<span class="op">=</span><span class="st">"SpeciesPoolR"</span><span class="op">)</span></span>
<span><span class="va">filtered_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_data.html">get_data</a></span><span class="op">(</span></span>
<span>   file <span class="op">=</span> <span class="va">f</span>,</span>
<span>   filter <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/substitute.html" class="external-link">quote</a></span><span class="op">(</span><span class="va">Kingdom</span> <span class="op">==</span> <span class="st">"Plantae"</span> <span class="op">&amp;</span> </span>
<span>                    <span class="va">Class</span> <span class="op">==</span> <span class="st">"Magnoliopsida"</span> <span class="op">&amp;</span> </span>
<span>                    <span class="va">Family</span> <span class="op">==</span> <span class="st">"Fabaceae"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>This will generate a dataset that can be used subsequently to count species presences and download species data as seen in table <a href="#tab:tablespecies">4.1</a></p>
<table class="table">
<colgroup>
<col width="13%">
<col width="8%">
<col width="14%">
<col width="14%">
<col width="8%">
<col width="9%">
<col width="10%">
<col width="21%">
</colgroup>
<thead><tr class="header">
<th align="left">redlist_2010</th>
<th align="left">Kingdom</th>
<th align="left">Phyllum</th>
<th align="left">Class</th>
<th align="left">Order</th>
<th align="left">Family</th>
<th align="left">Genus</th>
<th align="left">Species</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">NA</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Genista</td>
<td align="left">Genista tinctoria</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Trifolium</td>
<td align="left">Trifolium vesiculosum</td>
</tr>
<tr class="even">
<td align="left">LC</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sativa</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Lathyrus</td>
<td align="left">Lathyrus latifolius</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Anthyllis</td>
<td align="left">Anthyllis vulneraria</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Lathyrus</td>
<td align="left">Lathyrus japonicus</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia villosa</td>
</tr>
</tbody>
</table>
<p><span id="tab:tablespecies"></span>Table 4.1: Species that will be used to generate species pools</p>
</div>
<div class="section level3">
<h3 id="id_412-step-2-taxonomic-harmonization">4.1.2 Step 2: Taxonomic Harmonization<a class="anchor" aria-label="anchor" href="#id_412-step-2-taxonomic-harmonization"></a>
</h3>
<p>Next, you should perform taxonomic harmonization to ensure that the species names you use are recognized by the GBIF taxonomic backbone. This can be done using the Clean_Taxa function:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Clean_Species</span> <span class="op">&lt;-</span> <span class="fu">SpeciesPoolR</span><span class="fu">::</span><span class="fu"><a href="reference/Clean_Taxa.html">Clean_Taxa</a></span><span class="op">(</span><span class="va">filtered_data</span><span class="op">$</span><span class="va">Species</span><span class="op">)</span></span></code></pre></div>
<p>The resulting data frame, with harmonized species names, is shown in table <a href="#tab:cleantable">4.2</a></p>
<table style="width:100%;" class="table">
<colgroup>
<col width="12%">
<col width="12%">
<col width="6%">
<col width="12%">
<col width="4%">
<col width="7%">
<col width="8%">
<col width="4%">
<col width="5%">
<col width="6%">
<col width="12%">
<col width="4%">
</colgroup>
<thead><tr class="header">
<th align="left">Taxa</th>
<th align="left">matched_name2</th>
<th align="right">confidence</th>
<th align="left">canonicalName</th>
<th align="left">kingdom</th>
<th align="left">phylum</th>
<th align="left">class</th>
<th align="left">order</th>
<th align="left">family</th>
<th align="left">genus</th>
<th align="left">species</th>
<th align="left">rank</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Vicia sepium</td>
<td align="left">Vicia sepium</td>
<td align="right">99</td>
<td align="left">Vicia sepium</td>
<td align="left">Plantae</td>
<td align="left">Tracheophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sepium</td>
<td align="left">SPECIES</td>
</tr>
<tr class="even">
<td align="left">Genista tinctoria</td>
<td align="left">Genista tinctoria</td>
<td align="right">99</td>
<td align="left">Genista tinctoria</td>
<td align="left">Plantae</td>
<td align="left">Tracheophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Genista</td>
<td align="left">Genista tinctoria</td>
<td align="left">SPECIES</td>
</tr>
<tr class="odd">
<td align="left">Trifolium vesiculosum</td>
<td align="left">Trifolium vesiculosum</td>
<td align="right">99</td>
<td align="left">Trifolium vesiculosum</td>
<td align="left">Plantae</td>
<td align="left">Tracheophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Trifolium</td>
<td align="left">Trifolium vesiculosum</td>
<td align="left">SPECIES</td>
</tr>
<tr class="even">
<td align="left">Vicia sativa</td>
<td align="left">Vicia sativa</td>
<td align="right">97</td>
<td align="left">Vicia sativa</td>
<td align="left">Plantae</td>
<td align="left">Tracheophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sativa</td>
<td align="left">SPECIES</td>
</tr>
<tr class="odd">
<td align="left">Lathyrus latifolius</td>
<td align="left">Lathyrus latifolius</td>
<td align="right">98</td>
<td align="left">Lathyrus latifolius</td>
<td align="left">Plantae</td>
<td align="left">Tracheophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Lathyrus</td>
<td align="left">Lathyrus latifolius</td>
<td align="left">SPECIES</td>
</tr>
<tr class="even">
<td align="left">Anthyllis vulneraria</td>
<td align="left">Anthyllis vulneraria</td>
<td align="right">97</td>
<td align="left">Anthyllis vulneraria</td>
<td align="left">Plantae</td>
<td align="left">Tracheophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Anthyllis</td>
<td align="left">Anthyllis vulneraria</td>
<td align="left">SPECIES</td>
</tr>
<tr class="odd">
<td align="left">Lathyrus japonicus</td>
<td align="left">Lathyrus japonicus</td>
<td align="right">99</td>
<td align="left">Lathyrus japonicus</td>
<td align="left">Plantae</td>
<td align="left">Tracheophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Lathyrus</td>
<td align="left">Lathyrus japonicus</td>
<td align="left">SPECIES</td>
</tr>
<tr class="even">
<td align="left">Vicia villosa</td>
<td align="left">Vicia villosa</td>
<td align="right">97</td>
<td align="left">Vicia villosa</td>
<td align="left">Plantae</td>
<td align="left">Tracheophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia villosa</td>
<td align="left">SPECIES</td>
</tr>
</tbody>
</table>
<p><span id="tab:cleantable"></span>Table 4.2: Taxonomicallty harmonized dataset</p>
</div>
<div class="section level3">
<h3 id="id_413-step-3-counting-species-presences">4.1.3 Step 3: Counting Species Presences<a class="anchor" aria-label="anchor" href="#id_413-step-3-counting-species-presences"></a>
</h3>
<p>After harmonizing the species names, it’s important to obtain the number of occurrences of each species in your study area, especially if you plan to calculate rarity. You can do this using the <code>count_presences</code> function. This function allows you to filter occurrences by country or by a shapefile. Below is an example for Denmark:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Assuming Clean_Species is your data frame</span></span>
<span><span class="va">Count_DK</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/count_presences.html">count_presences</a></span><span class="op">(</span><span class="va">Clean_Species</span>, country <span class="op">=</span> <span class="st">"DK"</span><span class="op">)</span></span></code></pre></div>
<p>The resulting data frame of species presences in Denmark is shown in table <a href="#tab:tableCountDenmark">4.3</a></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">Count_DK</span>, caption <span class="op">=</span> <span class="st">"Counts of presences for the different species within Denmark"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">family</th>
<th align="left">genus</th>
<th align="left">species</th>
<th align="right">N</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sepium</td>
<td align="right">2897</td>
</tr>
<tr class="even">
<td align="left">Fabaceae</td>
<td align="left">Genista</td>
<td align="left">Genista tinctoria</td>
<td align="right">988</td>
</tr>
<tr class="odd">
<td align="left">Fabaceae</td>
<td align="left">Trifolium</td>
<td align="left">Trifolium vesiculosum</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sativa</td>
<td align="right">17379</td>
</tr>
<tr class="odd">
<td align="left">Fabaceae</td>
<td align="left">Lathyrus</td>
<td align="left">Lathyrus latifolius</td>
<td align="right">684</td>
</tr>
<tr class="even">
<td align="left">Fabaceae</td>
<td align="left">Anthyllis</td>
<td align="left">Anthyllis vulneraria</td>
<td align="right">8876</td>
</tr>
<tr class="odd">
<td align="left">Fabaceae</td>
<td align="left">Lathyrus</td>
<td align="left">Lathyrus japonicus</td>
<td align="right">3904</td>
</tr>
<tr class="even">
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia villosa</td>
<td align="right">243</td>
</tr>
</tbody>
</table>
<p><span id="tab:tableCountDenmark"></span>Table 4.3: Counts of presences for the different species within Denmark</p>
<p>Alternatively, you can filter by a specific region using a shapefile. For example, to count species presences within Aarhus commune:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">shp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/Aarhus.shp"</span>, package<span class="op">=</span><span class="st">"SpeciesPoolR"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Count_Aarhus</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/count_presences.html">count_presences</a></span><span class="op">(</span><span class="va">Clean_Species</span>, shapefile <span class="op">=</span> <span class="va">shp</span><span class="op">)</span></span></code></pre></div>
<p>The resulting data.frame for Aarhus commune is shown int table <a href="#tab:tableCountAarhus">4.4</a></p>
<table class="table">
<thead><tr class="header">
<th align="left">family</th>
<th align="left">genus</th>
<th align="left">species</th>
<th align="right">N</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sepium</td>
<td align="right">283</td>
</tr>
<tr class="even">
<td align="left">Fabaceae</td>
<td align="left">Genista</td>
<td align="left">Genista tinctoria</td>
<td align="right">27</td>
</tr>
<tr class="odd">
<td align="left">Fabaceae</td>
<td align="left">Trifolium</td>
<td align="left">Trifolium vesiculosum</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sativa</td>
<td align="right">467</td>
</tr>
<tr class="odd">
<td align="left">Fabaceae</td>
<td align="left">Lathyrus</td>
<td align="left">Lathyrus latifolius</td>
<td align="right">41</td>
</tr>
<tr class="even">
<td align="left">Fabaceae</td>
<td align="left">Anthyllis</td>
<td align="left">Anthyllis vulneraria</td>
<td align="right">153</td>
</tr>
<tr class="odd">
<td align="left">Fabaceae</td>
<td align="left">Lathyrus</td>
<td align="left">Lathyrus japonicus</td>
<td align="right">39</td>
</tr>
<tr class="even">
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia villosa</td>
<td align="right">10</td>
</tr>
</tbody>
</table>
<p><span id="tab:tableCountAarhus"></span>Table 4.4: Counts of presences for the different species within Aarhus commune</p>
<p>Now it is recommended to eliminate species that have no occurrences in the area, this is done automatically in the workflow version:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com" class="external-link">data.table</a></span><span class="op">)</span></span>
<span><span class="va">Count_Aarhus</span> <span class="op">&lt;-</span> <span class="va">Count_Aarhus</span><span class="op">[</span><span class="va">N</span> <span class="op">&gt;</span> <span class="fl">0</span>,<span class="op">]</span></span></code></pre></div>
<p>So that then we can retrieve the species presences using the function <code><a href="reference/get_presences.html">SpeciesPoolR::get_presences</a></code>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Presences</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_presences.html">get_presences</a></span><span class="op">(</span>species <span class="op">=</span> <span class="va">Count_Aarhus</span><span class="op">$</span><span class="va">species</span>, shapefile <span class="op">=</span> <span class="va">shp</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Geometry created: POLYGON ((10.401438 56.302419, 10.048024 56.355225, 9.886316 56.019928, 10.239729 55.966657, 10.401438 56.302419))"</span></span></code></pre></div>
<p>there we end up with 1075 presences for our 7 species.</p>
</div>
</div>
<div class="section level2">
<h2 id="id_42-creating-spatial-buffers-and-habitat-filtering">4.2 Creating Spatial Buffers and Habitat Filtering<a class="anchor" aria-label="anchor" href="#id_42-creating-spatial-buffers-and-habitat-filtering"></a>
</h2>
<div class="section level3">
<h3 id="id_421-step-1-creating-buffers-around-species-presences">4.2.1 Step 1 Creating Buffers Around Species Presences<a class="anchor" aria-label="anchor" href="#id_421-step-1-creating-buffers-around-species-presences"></a>
</h3>
<p>Once you have identified the species presences within your area of interest, the next step is to create spatial buffers around these occurrences. These buffers represent the potential dispersal range of each species, helping to assess areas where the species might establish itself given a specified dispersal distance.</p>
<p>To create these buffers, you’ll use a raster file as a template to rasterize the buffers and specify the distance (in meters) representing the species’ dispersal range.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Raster</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/LU_Aarhus.tif"</span>, package<span class="op">=</span><span class="st">"SpeciesPoolR"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">buffer500</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/make_buffer_rasterized.html">make_buffer_rasterized</a></span><span class="op">(</span><span class="va">Presences</span>, file <span class="op">=</span> <span class="va">Raster</span>, dist <span class="op">=</span> <span class="fl">500</span><span class="op">)</span></span></code></pre></div>
<p>In this example, the make_buffer_rasterized function generates a 500-meter buffer around each occurrence point in the Presences dataset. The function utilizes the provided raster file as a template for rasterizing these buffers.</p>
<p>The resulting buffer500 data frame indicates which raster cells are covered by the buffer for each species. Table <a href="#tab:showbuffer500">4.5</a> displays the first 10 observations of this data frame, providing a detailed view of the buffer’s overlap with raster cells, listing each cell and the corresponding species within that buffer.</p>
<table class="table">
<thead><tr class="header">
<th align="right">cell</th>
<th align="left">species</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">26</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="even">
<td align="right">27</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="odd">
<td align="right">28</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="even">
<td align="right">29</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="odd">
<td align="right">30</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="even">
<td align="right">161</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="odd">
<td align="right">162</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="even">
<td align="right">163</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="odd">
<td align="right">164</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="even">
<td align="right">165</td>
<td align="left">Vicia sepium</td>
</tr>
</tbody>
</table>
<p><span id="tab:showbuffer500"></span>Table 4.5: Raster cells within the 500-meter buffer of each species</p>
<p>This table provides a detailed view of how the buffer overlaps with the raster cells, listing each cell and the corresponding species present within that buffer.</p>
</div>
<div class="section level3">
<h3 id="id_422-step-2-habitat-filtering">4.2.2 Step 2: Habitat Filtering<a class="anchor" aria-label="anchor" href="#id_422-step-2-habitat-filtering"></a>
</h3>
<p>After creating the buffers, the next logical step is to filter these areas based on habitat suitability. This allows you to focus on specific land-use types or habitats where the species is more likely to thrive. Habitat filtering typically involves using raster data to refine or subset the buffer areas according to the desired habitat criteria.</p>
<div class="section level4">
<h4 id="id_4221-preparing-land-use-data">4.2.2.1 Preparing Land-Use Data<a class="anchor" aria-label="anchor" href="#id_4221-preparing-land-use-data"></a>
</h4>
<p>Before you can apply habitat filtering, you need to prepare a long-format land-use table that matches each raster cell to its corresponding habitat types. This is done using the generate_long_landuse_table function, which takes the path to your raster file and transforms it into a long-format data frame. The function also filters the data to include only those cells where the suitability value is 1 for at least one land-use type.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get path for habitat suitability</span></span>
<span><span class="va">HabSut</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/HabSut.tif"</span>, package <span class="op">=</span> <span class="st">"SpeciesPoolR"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Generate the long-format land-use table</span></span>
<span><span class="va">long_LU_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/generate_long_landuse_table.html">generate_long_landuse_table</a></span><span class="op">(</span>path <span class="op">=</span> <span class="va">HabSut</span><span class="op">)</span></span></code></pre></div>
<p>This is crucial for the next steps, the result is shown in table <a href="#tab:longtablehab">4.6</a>, as it links each raster cell to potential habitats, enabling you to match species occurrences to suitable environments within their buffer zones.</p>
<table class="table">
<thead><tr class="header">
<th align="right">cell</th>
<th align="left">Habitat</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">79</td>
<td align="left">OpenDryPoor</td>
</tr>
<tr class="even">
<td align="right">80</td>
<td align="left">OpenDryPoor</td>
</tr>
<tr class="odd">
<td align="right">81</td>
<td align="left">OpenDryPoor</td>
</tr>
<tr class="even">
<td align="right">82</td>
<td align="left">OpenDryPoor</td>
</tr>
<tr class="odd">
<td align="right">83</td>
<td align="left">OpenDryPoor</td>
</tr>
<tr class="even">
<td align="right">214</td>
<td align="left">OpenDryPoor</td>
</tr>
<tr class="odd">
<td align="right">215</td>
<td align="left">OpenDryPoor</td>
</tr>
<tr class="even">
<td align="right">216</td>
<td align="left">OpenDryPoor</td>
</tr>
<tr class="odd">
<td align="right">217</td>
<td align="left">OpenDryPoor</td>
</tr>
<tr class="even">
<td align="right">218</td>
<td align="left">OpenDryPoor</td>
</tr>
</tbody>
</table>
<p><span id="tab:longtablehab"></span>Table 4.6: First 10 observations of landuse suitability per cell</p>
</div>
<div class="section level4">
<h4 id="id_4222-applying-habitat-filtering">4.2.2.2 Applying Habitat Filtering<a class="anchor" aria-label="anchor" href="#id_4222-applying-habitat-filtering"></a>
</h4>
<p>Once you have the long-format land-use table, you can proceed with habitat filtering. To achieve this, you’ll use the <code>ModelAndPredictFunc</code>, which takes the presence data frame (e.g., Presences) obtained through the get_presences function and the land-use raster. This comprehensive function encompasses several critical steps:</p>
<p>1- <em>Grouping Data by Species</em>: The presence data is grouped by species using <code>group_split</code>, ensuring that each species is modeled individually.</p>
<p>2- <em>Sampling Land-Use Data</em>: For each species, land-use data is sampled at the presence points using the SampleLanduse function.</p>
<p>3- <em>Sampling Background Data</em>: Background points are also sampled from the same land-use raster, providing a contrast to the presence data.</p>
<p>4- <em>Modeling Habitat Suitability</em>: The presence and background data are combined and passed to the <code>ModelSpecies</code> function. This function fits a MaxEnt model to predict habitat suitability across the different land-use types.</p>
<p>5- <em>Predicting Suitability</em>: The fitted model is then used to predict habitat suitability for each species across all available land-use types.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Habitats</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/ModelAndPredictFunc.html">ModelAndPredictFunc</a></span><span class="op">(</span>DF <span class="op">=</span> <span class="va">Presences</span>, file <span class="op">=</span> <span class="va">Raster</span><span class="op">)</span></span></code></pre></div>
<p>The resulting Habitats data frame contains continuous suitability predictions for each species across various land-use types. Table <a href="#tab:tablespeciespred">4.7</a> shows the first 9 observations, illustrating the predicted habitat suitability scores for the first species in each land-use type.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">Habitats</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">9</span>,<span class="op">]</span>, caption <span class="op">=</span> <span class="st">"Predicted habitat suitability scores across various land-use types for the first species. The values represent continuous predictions, indicating the relative likelihood of species presence in each land-use category."</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">Landuse</th>
<th align="right">Pred</th>
<th align="left">species</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">OpenDryRich</td>
<td align="right">1.0000000</td>
<td align="left">Anthyllis vulneraria</td>
</tr>
<tr class="even">
<td align="left">OpenDryPoor</td>
<td align="right">1.0000000</td>
<td align="left">Anthyllis vulneraria</td>
</tr>
<tr class="odd">
<td align="left">ForestWetRich</td>
<td align="right">0.6838430</td>
<td align="left">Anthyllis vulneraria</td>
</tr>
<tr class="even">
<td align="left">OpenWetRich</td>
<td align="right">0.6838430</td>
<td align="left">Anthyllis vulneraria</td>
</tr>
<tr class="odd">
<td align="left">OpenWetPoor</td>
<td align="right">0.6838430</td>
<td align="left">Anthyllis vulneraria</td>
</tr>
<tr class="even">
<td align="left">Exclude</td>
<td align="right">0.5108840</td>
<td align="left">Anthyllis vulneraria</td>
</tr>
<tr class="odd">
<td align="left">ForestDryRich</td>
<td align="right">0.3207105</td>
<td align="left">Anthyllis vulneraria</td>
</tr>
<tr class="even">
<td align="left">ForestDryPoor</td>
<td align="right">0.2241332</td>
<td align="left">Anthyllis vulneraria</td>
</tr>
<tr class="odd">
<td align="left">Exclude</td>
<td align="right">0.6335459</td>
<td align="left">Genista tinctoria</td>
</tr>
</tbody>
</table>
<p><span id="tab:tablespeciespred"></span>Table 4.7: Predicted habitat suitability scores across various land-use types for the first species. The values represent continuous predictions, indicating the relative likelihood of species presence in each land-use category.</p>
</div>
<div class="section level4">
<h4 id="id_4223-maximum-entropy-model-for-land-use-selection">4.2.2.3 Maximum Entropy Model for Land Use Selection<a class="anchor" aria-label="anchor" href="#id_4223-maximum-entropy-model-for-land-use-selection"></a>
</h4>
<p>To assess the land use preferences of the target species, we employed a customized Maximum Entropy (MaxEnt) model. This model was developed not as a traditional Species Distribution Model (SDM), but as a lookup table to evaluate the relative selection of land use types. We began by identifying species presence locations and extracting the corresponding land use categories. A minimum convex polygon (MCP) was constructed around these presence points and expanded by 15% to define the background area. Random sampling within this background area provided a distribution of land use types for comparison. The MaxEnt model was then built to evaluate the ratio of the probability density of land use types at presence locations (P(z)) to that at background locations (Q(z)). This ratio, P(z)/Q(z), allowed us to identify land use types that were disproportionately selected by the species. The model utilized a cloglog transformation to convert these ratios into probabilities, providing insights into the species’ land use preferences. The final output was a predictive model indicating the probability of land use selection by the species, highlighting areas where the observed land use differed from that expected based on background availability.</p>
</div>
</div>
<div class="section level3">
<h3 id="id_423-step-3-generating-habitat-suitability-thresholds">4.2.3 Step 3: Generating Habitat Suitability Thresholds<a class="anchor" aria-label="anchor" href="#id_423-step-3-generating-habitat-suitability-thresholds"></a>
</h3>
<p>While continuous predictions provide a detailed picture of habitat suitability, it is often useful to classify these predictions into binary suitability thresholds. Thresholds can help determine areas where species presence is more likely or unlikely based on habitat preferences.</p>
<p>The create_thresholds function facilitates this by generating thresholds based on the modeled land-use preferences, using the 90th, 95th, and 99th percentiles of the predicted suitability values. These thresholds represent the commission rates, helping to define the probability cutoff above which a land-use type is considered suitable for a species.</p>
<p>Here’s how you can generate these thresholds for the species in your dataset:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Thresholds</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/create_thresholds.html">create_thresholds</a></span><span class="op">(</span>Model <span class="op">=</span> <span class="va">Habitats</span>, reference <span class="op">=</span> <span class="va">Presences</span>, file <span class="op">=</span> <span class="va">Raster</span><span class="op">)</span></span></code></pre></div>
<p>This will generate de data set with the threshold for the comission rates of 90, 95 and 99th percentile for each species that can be seen in Table <a href="#tab:thresholdtables">4.8</a>.</p>
<table class="table">
<thead><tr class="header">
<th align="left">species</th>
<th align="right">Thres_99</th>
<th align="right">Thres_95</th>
<th align="right">Thres_90</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Anthyllis vulneraria</td>
<td align="right">0.511</td>
<td align="right">0.511</td>
<td align="right">0.511</td>
</tr>
<tr class="even">
<td align="left">Genista tinctoria</td>
<td align="right">0.634</td>
<td align="right">0.634</td>
<td align="right">0.634</td>
</tr>
<tr class="odd">
<td align="left">Lathyrus japonicus</td>
<td align="right">0.407</td>
<td align="right">0.407</td>
<td align="right">0.407</td>
</tr>
<tr class="even">
<td align="left">Lathyrus latifolius</td>
<td align="right">0.634</td>
<td align="right">0.634</td>
<td align="right">0.634</td>
</tr>
<tr class="odd">
<td align="left">Vicia sativa</td>
<td align="right">0.405</td>
<td align="right">0.405</td>
<td align="right">0.405</td>
</tr>
<tr class="even">
<td align="left">Vicia sepium</td>
<td align="right">0.294</td>
<td align="right">0.294</td>
<td align="right">0.294</td>
</tr>
<tr class="odd">
<td align="left">Vicia villosa</td>
<td align="right">0.633</td>
<td align="right">0.633</td>
<td align="right">0.633</td>
</tr>
</tbody>
</table>
<p><span id="tab:thresholdtables"></span>Table 4.8: Threshold based on commission rate for the species that are used above</p>
<p>This step produces a data frame containing the thresholds for each species, which can then be used to classify habitat suitability into binary categories, helping you to identify core habitats or areas of higher conservation value.</p>
<p>After we have the continuous thresholds we can generate a lookup table to see which species can inhabit in each landuse type</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">LookupTable</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/Generate_Lookup.html">Generate_Lookup</a></span><span class="op">(</span>Model <span class="op">=</span> <span class="va">Habitats</span>, Thresholds <span class="op">=</span> <span class="va">Thresholds</span><span class="op">)</span></span></code></pre></div>
<p>This creates Table <a href="#tab:lookuptab">4.9</a>, notice how it only shows for each species which habitats are available not the ones that are not.</p>
<table class="table">
<thead><tr class="header">
<th align="left">species</th>
<th align="left">Landuse</th>
<th align="right">Pres</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Anthyllis vulneraria</td>
<td align="left">OpenDryRich</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Anthyllis vulneraria</td>
<td align="left">OpenDryPoor</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">Anthyllis vulneraria</td>
<td align="left">ForestWetRich</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Anthyllis vulneraria</td>
<td align="left">OpenWetRich</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">Anthyllis vulneraria</td>
<td align="left">OpenWetPoor</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Lathyrus japonicus</td>
<td align="left">OpenDryPoor</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">Vicia sativa</td>
<td align="left">OpenDryPoor</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Vicia sativa</td>
<td align="left">OpenDryRich</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">Vicia sativa</td>
<td align="left">OpenWetPoor</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Vicia sativa</td>
<td align="left">OpenWetRich</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">Vicia sativa</td>
<td align="left">ForestWetRich</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Vicia sativa</td>
<td align="left">ForestDryRich</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">Vicia sepium</td>
<td align="left">ForestWetRich</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Vicia sepium</td>
<td align="left">ForestDryRich</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">Vicia sepium</td>
<td align="left">OpenDryPoor</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Vicia sepium</td>
<td align="left">OpenWetPoor</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">Vicia sepium</td>
<td align="left">OpenWetRich</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Vicia sepium</td>
<td align="left">OpenDryRich</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<p><span id="tab:lookuptab"></span>Table 4.9: dummy variable that shows which species can inhabit each habitat type</p>
</div>
<div class="section level3">
<h3 id="id_424-step-4-generating-final-species-presences">4.2.4 Step 4: Generating Final Species Presences<a class="anchor" aria-label="anchor" href="#id_424-step-4-generating-final-species-presences"></a>
</h3>
<p>In this final step, we apply the <code>make_final_presences</code> function to filter the buffered species presences. This filtering process is done in three stages:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Lookup Table Filtering:</strong> The function first ensures that each species is only considered in habitats where it can persist based on the species-habitat suitability mappings in the lookup table.</p></li>
<li><p><strong>Land-Use Table Filtering:</strong> Next, it filters these suitable habitats to include only those cells where the specific habitat type could exist, based on the long-format land-use table.</p></li>
<li><p><strong>Buffer Zone Filtering:</strong> Finally, it restricts the potential species occurrences to areas where the species is likely to disperse, as indicated by the spatial buffers generated around species presence points.</p></li>
</ol>
<p>The result is a highly refined dataset that specifies, for each species, the exact cells and habitat types where it can potentially occur, combining habitat suitability, land-use distribution, and species dispersal capability.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">final_presences</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/make_final_presences.html">make_final_presences</a></span><span class="op">(</span>Long_LU_table <span class="op">=</span> <span class="va">long_LU_table</span>, </span>
<span>                                        Long_Buffer_gbif <span class="op">=</span> <span class="va">buffer500</span>,</span>
<span>                                        LookUpTable <span class="op">=</span> <span class="va">LookupTable</span><span class="op">)</span></span></code></pre></div>
<p>The resulting <code>final_presences</code> table provides detailed information on the potential distribution of each species. It specifies which cells and habitats are suitable for each species, ensuring that only the most plausible locations are considered. In table @(tab:finalpresences), you can see the first 15 observations from this final dataset, which represent the potential habitats where each species could thrive, whereas in table @(tab:summaryfinalpresences), you can see a summary of the number of cells that each species could thrive on each habitat type.</p>
<table class="table">
<thead><tr class="header">
<th align="right">cell</th>
<th align="left">species</th>
<th align="left">Landuse</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">1018</td>
<td align="left">Anthyllis vulneraria</td>
<td align="left">OpenDryRich</td>
</tr>
<tr class="even">
<td align="right">1557</td>
<td align="left">Anthyllis vulneraria</td>
<td align="left">OpenDryRich</td>
</tr>
<tr class="odd">
<td align="right">1825</td>
<td align="left">Anthyllis vulneraria</td>
<td align="left">OpenDryRich</td>
</tr>
<tr class="even">
<td align="right">2093</td>
<td align="left">Anthyllis vulneraria</td>
<td align="left">OpenDryRich</td>
</tr>
<tr class="odd">
<td align="right">2215</td>
<td align="left">Anthyllis vulneraria</td>
<td align="left">OpenDryRich</td>
</tr>
<tr class="even">
<td align="right">2216</td>
<td align="left">Anthyllis vulneraria</td>
<td align="left">OpenDryRich</td>
</tr>
<tr class="odd">
<td align="right">2218</td>
<td align="left">Anthyllis vulneraria</td>
<td align="left">OpenDryRich</td>
</tr>
<tr class="even">
<td align="right">2351</td>
<td align="left">Anthyllis vulneraria</td>
<td align="left">OpenDryRich</td>
</tr>
<tr class="odd">
<td align="right">2352</td>
<td align="left">Anthyllis vulneraria</td>
<td align="left">OpenDryRich</td>
</tr>
<tr class="even">
<td align="right">2353</td>
<td align="left">Anthyllis vulneraria</td>
<td align="left">OpenDryRich</td>
</tr>
<tr class="odd">
<td align="right">2354</td>
<td align="left">Anthyllis vulneraria</td>
<td align="left">OpenDryRich</td>
</tr>
<tr class="even">
<td align="right">2486</td>
<td align="left">Anthyllis vulneraria</td>
<td align="left">OpenDryRich</td>
</tr>
<tr class="odd">
<td align="right">2488</td>
<td align="left">Anthyllis vulneraria</td>
<td align="left">OpenDryRich</td>
</tr>
<tr class="even">
<td align="right">2489</td>
<td align="left">Anthyllis vulneraria</td>
<td align="left">OpenDryRich</td>
</tr>
<tr class="odd">
<td align="right">2620</td>
<td align="left">Anthyllis vulneraria</td>
<td align="left">OpenDryRich</td>
</tr>
</tbody>
</table>
<p><span id="tab:finalpresences"></span>Table 4.10: First 15 rows of the final presences dataset, showing the cells and land-use types where each species can potentially occur</p>
<table class="table">
<thead><tr class="header">
<th align="left">Landuse</th>
<th align="left">species</th>
<th align="right">N</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">OpenDryRich</td>
<td align="left">Anthyllis vulneraria</td>
<td align="right">802</td>
</tr>
<tr class="even">
<td align="left">ForestWetRich</td>
<td align="left">Anthyllis vulneraria</td>
<td align="right">512</td>
</tr>
<tr class="odd">
<td align="left">OpenWetRich</td>
<td align="left">Anthyllis vulneraria</td>
<td align="right">512</td>
</tr>
<tr class="even">
<td align="left">ForestDryRich</td>
<td align="left">Vicia sepium</td>
<td align="right">254</td>
</tr>
<tr class="odd">
<td align="left">OpenDryRich</td>
<td align="left">Vicia sepium</td>
<td align="right">254</td>
</tr>
<tr class="even">
<td align="left">ForestWetRich</td>
<td align="left">Vicia sepium</td>
<td align="right">141</td>
</tr>
<tr class="odd">
<td align="left">OpenWetRich</td>
<td align="left">Vicia sepium</td>
<td align="right">141</td>
</tr>
<tr class="even">
<td align="left">OpenWetPoor</td>
<td align="left">Anthyllis vulneraria</td>
<td align="right">126</td>
</tr>
<tr class="odd">
<td align="left">OpenDryPoor</td>
<td align="left">Anthyllis vulneraria</td>
<td align="right">70</td>
</tr>
<tr class="even">
<td align="left">OpenDryPoor</td>
<td align="left">Lathyrus japonicus</td>
<td align="right">47</td>
</tr>
<tr class="odd">
<td align="left">OpenWetPoor</td>
<td align="left">Vicia sepium</td>
<td align="right">31</td>
</tr>
<tr class="even">
<td align="left">ForestDryRich</td>
<td align="left">Vicia sativa</td>
<td align="right">26</td>
</tr>
<tr class="odd">
<td align="left">OpenDryRich</td>
<td align="left">Vicia sativa</td>
<td align="right">26</td>
</tr>
<tr class="even">
<td align="left">ForestWetRich</td>
<td align="left">Vicia sativa</td>
<td align="right">18</td>
</tr>
<tr class="odd">
<td align="left">OpenWetRich</td>
<td align="left">Vicia sativa</td>
<td align="right">18</td>
</tr>
<tr class="even">
<td align="left">OpenDryPoor</td>
<td align="left">Vicia sepium</td>
<td align="right">16</td>
</tr>
<tr class="odd">
<td align="left">OpenWetPoor</td>
<td align="left">Vicia sativa</td>
<td align="right">7</td>
</tr>
</tbody>
</table>
<p><span id="tab:summaryfinalpresences"></span>Table 4.11: Summary of number of cells that each species can thrive in for each habitat type</p>
</div>
</div>
<div class="section level2">
<h2 id="id_43-generating-summary-biodiversity-statistics">4.3 Generating summary biodiversity statistics<a class="anchor" aria-label="anchor" href="#id_43-generating-summary-biodiversity-statistics"></a>
</h2>
<div class="section level3">
<h3 id="id_431-step-1-generating-phylogenetic-diversity-metrics">4.3.1 Step 1 Generating Phylogenetic diversity metrics<a class="anchor" aria-label="anchor" href="#id_431-step-1-generating-phylogenetic-diversity-metrics"></a>
</h3>
<p>In order to generate Phylogenetic Diversity measures, the first step is to generate a phylogenetic tree with the species we have, for that we will use the V.Phylomaker package function <code>phylo.maker</code>based on the megaphylogeny of vascular plants (Jin and Qian 2019; Zanne et al. 2014), this means that we can only use this functions in species pools of plants.</p>
<p>In this case we use the <code>generate_tree</code> from SpeciesPoolR to do so:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/generate_tree.html">generate_tree</a></span><span class="op">(</span><span class="va">Count_Aarhus</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "All species in sp.list are present on tree."</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level1">
<h1 id="id_5-running-the-speciespoolr-workflow">5 Running the SpeciesPoolR Workflow<a class="anchor" aria-label="anchor" href="#id_5-running-the-speciespoolr-workflow"></a>
</h1>
<p>If you prefer to automate the process and run the <code>SpeciesPoolR</code> workflow as a pipeline, you can use the <code>run_workflow</code> function. This function sets up a <code>targets</code> workflow that sequentially executes the steps for cleaning species data, counting species presences, and performing spatial analysis. This approach is especially useful for larger datasets or when you want to ensure reproducibility.</p>
<p>To run the workflow, you can use the following code. We’ll use the same species filter as before, focusing on the <code>Plantae</code> kingdom, <code>Magnoliopsida</code> class, and <code>Fabaceae</code> family. Additionally, we’ll focus on the Aarhus commune using a shapefile.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">shp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/Aarhus.shp"</span>, package <span class="op">=</span> <span class="st">"SpeciesPoolR"</span><span class="op">)</span></span>
<span><span class="va">Raster</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/LU_Aarhus.tif"</span>, package<span class="op">=</span><span class="st">"SpeciesPoolR"</span><span class="op">)</span></span>
<span><span class="va">HabSut</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/HabSut.tif"</span>, package <span class="op">=</span> <span class="st">"SpeciesPoolR"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="reference/run_workflow.html">run_workflow</a></span><span class="op">(</span></span>
<span>  file_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/Species_List.csv"</span>, package <span class="op">=</span> <span class="st">"SpeciesPoolR"</span><span class="op">)</span>,</span>
<span>  filter <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/substitute.html" class="external-link">quote</a></span><span class="op">(</span><span class="va">Kingdom</span> <span class="op">==</span> <span class="st">"Plantae"</span> <span class="op">&amp;</span> <span class="va">Class</span> <span class="op">==</span> <span class="st">"Magnoliopsida"</span> <span class="op">&amp;</span> <span class="va">Family</span> <span class="op">==</span> <span class="st">"Fabaceae"</span><span class="op">)</span>,</span>
<span>  shapefile <span class="op">=</span> <span class="va">shp</span>,</span>
<span>  dist <span class="op">=</span> <span class="fl">500</span>,</span>
<span>  rastertemp <span class="op">=</span> <span class="va">Raster</span>,</span>
<span>  rasterLU <span class="op">=</span> <span class="va">Raster</span>,</span>
<span>  LanduseSuitability <span class="op">=</span> <span class="va">HabSut</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; ▶ dispatched target shp</span></span>
<span><span class="co">#&gt; ▶ dispatched target Raster</span></span>
<span><span class="co">#&gt; ● completed target shp [8.94 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched target Landuses</span></span>
<span><span class="co">#&gt; ● completed target Landuses [0 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched target file</span></span>
<span><span class="co">#&gt; ● completed target file [0 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched target data</span></span>
<span><span class="co">#&gt; ● completed target Raster [8.884 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched target Landusesuitability</span></span>
<span><span class="co">#&gt; ● completed target Landusesuitability [0 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched target Long_LU_table</span></span>
<span><span class="co">#&gt; ● completed target Long_LU_table [0.113 seconds]</span></span>
<span><span class="co">#&gt; ● completed target data [0.665 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched target Clean</span></span>
<span><span class="co">#&gt; ● completed target Clean [1.531 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Count_Presences_33538e94b3809372</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Count_Presences_52d72a5ad405e933</span></span>
<span><span class="co">#&gt; ● completed branch Count_Presences_33538e94b3809372 [0.229 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Count_Presences_e70f77d9439a4770</span></span>
<span><span class="co">#&gt; ● completed branch Count_Presences_e70f77d9439a4770 [0.855 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Count_Presences_dea4ef8633a449a1</span></span>
<span><span class="co">#&gt; ● completed branch Count_Presences_52d72a5ad405e933 [1.168 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Count_Presences_69210fc440d13855</span></span>
<span><span class="co">#&gt; ● completed branch Count_Presences_69210fc440d13855 [0.34 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Count_Presences_a61be030e01ebaf5</span></span>
<span><span class="co">#&gt; ● completed branch Count_Presences_dea4ef8633a449a1 [0.398 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Count_Presences_974105e269324d3e</span></span>
<span><span class="co">#&gt; ● completed branch Count_Presences_974105e269324d3e [0.121 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Count_Presences_37d1f8d5f74d852c</span></span>
<span><span class="co">#&gt; ● completed branch Count_Presences_a61be030e01ebaf5 [0.14 seconds]</span></span>
<span><span class="co">#&gt; ● completed branch Count_Presences_37d1f8d5f74d852c [0.623 seconds]</span></span>
<span><span class="co">#&gt; ● completed pattern Count_Presences</span></span>
<span><span class="co">#&gt; ▶ dispatched target More_than_zero</span></span>
<span><span class="co">#&gt; ● completed target More_than_zero [0.002 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Presences_c112b37cd15959d6</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Presences_af64bac105a08467</span></span>
<span><span class="co">#&gt; ● completed branch Presences_af64bac105a08467 [0.905 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch ModelAndPredict_0e19b8cb545404d2</span></span>
<span><span class="co">#&gt; ● completed branch Presences_c112b37cd15959d6 [1.935 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch ModelAndPredict_626a53b08dfe709d</span></span>
<span><span class="co">#&gt; ● completed branch ModelAndPredict_0e19b8cb545404d2 [1.476 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Presences_daf8d6353bc80f0c</span></span>
<span><span class="co">#&gt; ● completed branch Presences_daf8d6353bc80f0c [2.269 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch ModelAndPredict_edb09c8ec5c9a988</span></span>
<span><span class="co">#&gt; ● completed branch ModelAndPredict_626a53b08dfe709d [20.212 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Presences_310adeccf6b44725</span></span>
<span><span class="co">#&gt; ● completed branch Presences_310adeccf6b44725 [1.195 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch ModelAndPredict_b226446ac3154351</span></span>
<span><span class="co">#&gt; ● completed branch ModelAndPredict_b226446ac3154351 [6.505 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Presences_e65f4227e8299cc4</span></span>
<span><span class="co">#&gt; ● completed branch ModelAndPredict_edb09c8ec5c9a988 [25.631 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Presences_d4b9dc68293bd5b2</span></span>
<span><span class="co">#&gt; ● completed branch Presences_d4b9dc68293bd5b2 [0.869 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch ModelAndPredict_cae8301e59fc4e01</span></span>
<span><span class="co">#&gt; ● completed branch Presences_e65f4227e8299cc4 [1.397 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch ModelAndPredict_0a8436ee3d4f2644</span></span>
<span><span class="co">#&gt; ● completed branch ModelAndPredict_cae8301e59fc4e01 [0.936 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Presences_88937156c1302a12</span></span>
<span><span class="co">#&gt; ● completed branch Presences_88937156c1302a12 [1.18 seconds]</span></span>
<span><span class="co">#&gt; ● completed pattern Presences</span></span>
<span><span class="co">#&gt; ▶ dispatched branch ModelAndPredict_a0190cbfdf5f6f1f</span></span>
<span><span class="co">#&gt; ● completed branch ModelAndPredict_a0190cbfdf5f6f1f [0.392 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched target Phylo_Tree</span></span>
<span><span class="co">#&gt; ● completed branch ModelAndPredict_0a8436ee3d4f2644 [11.237 seconds]</span></span>
<span><span class="co">#&gt; ● completed pattern ModelAndPredict</span></span>
<span><span class="co">#&gt; ▶ dispatched target Thresholds</span></span>
<span><span class="co">#&gt; ● completed target Thresholds [0.764 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched target LookUpTable</span></span>
<span><span class="co">#&gt; ● completed target LookUpTable [0.013 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched target rarity_weight</span></span>
<span><span class="co">#&gt; ● completed target rarity_weight [0.004 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch buffer_0e19b8cb545404d2</span></span>
<span><span class="co">#&gt; ● completed branch buffer_0e19b8cb545404d2 [0.152 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Final_Presences_344cc771c9264c2e</span></span>
<span><span class="co">#&gt; ● completed branch Final_Presences_344cc771c9264c2e [0.156 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch buffer_626a53b08dfe709d</span></span>
<span><span class="co">#&gt; ● completed branch buffer_626a53b08dfe709d [0.23 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Final_Presences_ebf0f62f14548a82</span></span>
<span><span class="co">#&gt; ● completed branch Final_Presences_ebf0f62f14548a82 [0.293 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch buffer_edb09c8ec5c9a988</span></span>
<span><span class="co">#&gt; ● completed branch buffer_edb09c8ec5c9a988 [0.146 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Final_Presences_6f1885e07badc469</span></span>
<span><span class="co">#&gt; ● completed branch Final_Presences_6f1885e07badc469 [0.077 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch buffer_b226446ac3154351</span></span>
<span><span class="co">#&gt; ● completed branch buffer_b226446ac3154351 [0.058 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Final_Presences_35ecd9eff835718c</span></span>
<span><span class="co">#&gt; ● completed branch Final_Presences_35ecd9eff835718c [0.076 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch buffer_cae8301e59fc4e01</span></span>
<span><span class="co">#&gt; ● completed branch buffer_cae8301e59fc4e01 [0.072 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Final_Presences_5224d468624d4ebb</span></span>
<span><span class="co">#&gt; ● completed branch Final_Presences_5224d468624d4ebb [0.031 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch buffer_0a8436ee3d4f2644</span></span>
<span><span class="co">#&gt; ● completed branch buffer_0a8436ee3d4f2644 [0.086 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Final_Presences_af0c167a6a4b9998</span></span>
<span><span class="co">#&gt; ● completed branch Final_Presences_af0c167a6a4b9998 [0.162 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch buffer_a0190cbfdf5f6f1f</span></span>
<span><span class="co">#&gt; ● completed branch buffer_a0190cbfdf5f6f1f [0.06 seconds]</span></span>
<span><span class="co">#&gt; ● completed pattern buffer</span></span>
<span><span class="co">#&gt; ▶ dispatched branch Final_Presences_d203d619aa280fd1</span></span>
<span><span class="co">#&gt; ● completed branch Final_Presences_d203d619aa280fd1 [0.039 seconds]</span></span>
<span><span class="co">#&gt; ● completed pattern Final_Presences</span></span>
<span><span class="co">#&gt; ▶ dispatched target unique_habitats</span></span>
<span><span class="co">#&gt; ● completed target unique_habitats [0 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch rarity_405e1cf7d36edc08</span></span>
<span><span class="co">#&gt; ● completed branch rarity_405e1cf7d36edc08 [0.209 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch output_Rarity_2cf706eb0499c812</span></span>
<span><span class="co">#&gt; ● completed branch output_Rarity_2cf706eb0499c812 [0.153 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched target unique_species</span></span>
<span><span class="co">#&gt; ● completed target unique_species [0.002 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch export_presences_b7bf78e1c1a430c9</span></span>
<span><span class="co">#&gt; ● completed branch export_presences_b7bf78e1c1a430c9 [4.809 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch rarity_77d9a26761e4a007</span></span>
<span><span class="co">#&gt; ● completed branch rarity_77d9a26761e4a007 [0.669 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch output_Rarity_261466b6f238f521</span></span>
<span><span class="co">#&gt; ● completed branch output_Rarity_261466b6f238f521 [1.779 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch rarity_47ea97700de70215</span></span>
<span><span class="co">#&gt; ● completed branch rarity_47ea97700de70215 [0.365 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch output_Rarity_c2db253b33fcf9d9</span></span>
<span><span class="co">#&gt; ● completed branch output_Rarity_c2db253b33fcf9d9 [0.236 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch rarity_f4a6e9a8f4837219</span></span>
<span><span class="co">#&gt; ● completed branch rarity_f4a6e9a8f4837219 [0.568 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch output_Rarity_c1388be691e0be60</span></span>
<span><span class="co">#&gt; ● completed branch output_Rarity_c1388be691e0be60 [0.159 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch rarity_bee04486eb86e311</span></span>
<span><span class="co">#&gt; ● completed branch rarity_bee04486eb86e311 [0.321 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch output_Rarity_7d8e043127d47e71</span></span>
<span><span class="co">#&gt; ● completed branch output_Rarity_7d8e043127d47e71 [0.146 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch rarity_fcb1676d3b2f6824</span></span>
<span><span class="co">#&gt; ● completed branch rarity_fcb1676d3b2f6824 [0.518 seconds]</span></span>
<span><span class="co">#&gt; ● completed pattern rarity</span></span>
<span><span class="co">#&gt; ▶ dispatched branch output_Rarity_c8f99e24ce7afc52</span></span>
<span><span class="co">#&gt; ● completed branch output_Rarity_c8f99e24ce7afc52 [0.164 seconds]</span></span>
<span><span class="co">#&gt; ● completed pattern output_Rarity</span></span>
<span><span class="co">#&gt; ▶ dispatched branch export_presences_9cb7df6f909cc656</span></span>
<span><span class="co">#&gt; ● completed branch export_presences_9cb7df6f909cc656 [1.54 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch export_presences_e0501a6e2e4e8857</span></span>
<span><span class="co">#&gt; ● completed branch export_presences_e0501a6e2e4e8857 [0.367 seconds]</span></span>
<span><span class="co">#&gt; ● completed pattern export_presences</span></span>
<span><span class="co">#&gt; ● completed target Phylo_Tree [54.599 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch PhyloDiversity_405e1cf7d36edc08</span></span>
<span><span class="co">#&gt; ▶ dispatched branch PhyloDiversity_77d9a26761e4a007</span></span>
<span><span class="co">#&gt; ● completed branch PhyloDiversity_405e1cf7d36edc08 [0.634 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch output_Richness_0fed6773594c8b8e</span></span>
<span><span class="co">#&gt; ● completed branch PhyloDiversity_77d9a26761e4a007 [0.716 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch output_Richness_9080f2064a157266</span></span>
<span><span class="co">#&gt; ● completed branch output_Richness_0fed6773594c8b8e [0.116 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch PhyloDiversity_47ea97700de70215</span></span>
<span><span class="co">#&gt; ● completed branch output_Richness_9080f2064a157266 [0.113 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch PhyloDiversity_f4a6e9a8f4837219</span></span>
<span><span class="co">#&gt; ● completed branch PhyloDiversity_47ea97700de70215 [0.068 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch output_Richness_5144605bf3a4ac38</span></span>
<span><span class="co">#&gt; ● completed branch PhyloDiversity_f4a6e9a8f4837219 [0.72 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch output_Richness_d34599733447ab6e</span></span>
<span><span class="co">#&gt; ● completed branch output_Richness_5144605bf3a4ac38 [0.581 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch PhyloDiversity_bee04486eb86e311</span></span>
<span><span class="co">#&gt; ● completed branch output_Richness_d34599733447ab6e [0.065 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch PhyloDiversity_fcb1676d3b2f6824</span></span>
<span><span class="co">#&gt; ● completed branch PhyloDiversity_bee04486eb86e311 [0.094 seconds]</span></span>
<span><span class="co">#&gt; ▶ dispatched branch output_Richness_c0ee68bb087b85bd</span></span>
<span><span class="co">#&gt; ● completed branch PhyloDiversity_fcb1676d3b2f6824 [0.599 seconds]</span></span>
<span><span class="co">#&gt; ● completed pattern PhyloDiversity</span></span>
<span><span class="co">#&gt; ▶ dispatched branch output_Richness_32f01d33f95b1de0</span></span>
<span><span class="co">#&gt; ● completed branch output_Richness_c0ee68bb087b85bd [0.087 seconds]</span></span>
<span><span class="co">#&gt; ● completed branch output_Richness_32f01d33f95b1de0 [0.09 seconds]</span></span>
<span><span class="co">#&gt; ● completed pattern output_Richness</span></span>
<span><span class="co">#&gt; ▶ ended pipeline [2.305 minutes]</span></span>
<span><span class="co">#&gt; Warning message:</span></span>
<span><span class="co">#&gt; 10 targets produced warnings. Run targets::tar_meta(fields = warnings, complete_only = TRUE) for the messages.</span></span></code></pre></div>
<p><img src="reference/figures/README-run_workflow-1.png" width="100%"></p>
<div class="section level2">
<h2 id="id_51-how-it-works">5.1 How It Works<a class="anchor" aria-label="anchor" href="#id_51-how-it-works"></a>
</h2>
<p>The <code>run_workflow</code> function creates a pipeline that:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Reads the data</strong> from the specified file path.</p></li>
<li><p><strong>Filters the data</strong> using the provided filter expression.</p></li>
<li><p><strong>Cleans the species names</strong> to match the GBIF taxonomic backbone.</p></li>
<li><p><strong>Counts the species presences</strong> within the specified geographic area (in this case, Aarhus).</p></li>
<li><p><strong>Generates a buffer</strong> around the species presences within the specified distance, using a template raster.</p></li>
<li><p><strong>Prepares the land-use data</strong> by generating a long-format table that matches each raster cell to its corresponding habitat types.</p></li>
<li><p><strong>Predicts habitat suitability</strong> for each species across different land-use types using the <code>ModelAndPredictFunc</code>, which models habitat preferences and provides continuous predictions.</p></li>
<li><p><strong>Generates habitat suitability thresholds</strong> for each species based on the predicted suitability scores, using the <code>create_thresholds</code> function to define the 90th, 95th, and 99th percentile thresholds.</p></li>
<li><p><strong>Builds a lookup table</strong> to determine the land-use types each species can inhabit based on the thresholds.</p></li>
<li><p><strong>Generates the final species presences</strong> by filtering the buffered presences according to both the lookup table and the long land-use table, ensuring each species’ potential distribution is consistent with its habitat preferences.</p></li>
<li><p><strong>Export final presences as csv</strong> the final presences of each species and landuse are exported to the folder <code>Field_Final_Presences</code> as csv files.</p></li>
<li><p><strong>Calculates rarity weights</strong> Calculates a Rarity weight for the counts of species that have more than zero presences.</p></li>
<li><p><strong>Calculates Rarity</strong> Calculates and index of relative rarity for each cell on each potential landuse for the cell.</p></li>
<li><p><strong>Generates a phylogenetic tree</strong> for the species in the species list, using the <code>generate_tree</code> function.</p></li>
<li><p><strong>Calculates phylogenetic phylogenetic diversity</strong> for each cell in each potential landuse.</p></li>
<li><p><strong>Generates a visual representation</strong> of the workflow if <code>plot = TRUE</code>.</p></li>
</ol>
<p>You can monitor the progress of the workflow and visualize the dependencies between steps using targets::tar_visnetwork(). The result will be similar to running the steps manually but with the added benefits of parallel execution and reproducibility.</p>
<p>This automated approach allows you to streamline your analysis and ensures that all steps are consistently applied to your data. It also makes it easier to rerun the workflow with different parameters or datasets.</p>
<p>The files generated are</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span>path <span class="op">=</span> <span class="st">"Results/"</span>, recursive <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "Rarity/Rarity_ForestDryRich.tfw"     "Rarity/Rarity_ForestDryRich.tif"    </span></span>
<span><span class="co">#&gt;  [3] "Rarity/Rarity_ForestWetRich.tfw"     "Rarity/Rarity_ForestWetRich.tif"    </span></span>
<span><span class="co">#&gt;  [5] "Rarity/Rarity_OpenDryPoor.tfw"       "Rarity/Rarity_OpenDryPoor.tif"      </span></span>
<span><span class="co">#&gt;  [7] "Rarity/Rarity_OpenDryRich.tfw"       "Rarity/Rarity_OpenDryRich.tif"      </span></span>
<span><span class="co">#&gt;  [9] "Rarity/Rarity_OpenWetPoor.tfw"       "Rarity/Rarity_OpenWetPoor.tif"      </span></span>
<span><span class="co">#&gt; [11] "Rarity/Rarity_OpenWetRich.tfw"       "Rarity/Rarity_OpenWetRich.tif"      </span></span>
<span><span class="co">#&gt; [13] "Richness/Richness_ForestDryRich.tfw" "Richness/Richness_ForestDryRich.tif"</span></span>
<span><span class="co">#&gt; [15] "Richness/Richness_ForestWetRich.tfw" "Richness/Richness_ForestWetRich.tif"</span></span>
<span><span class="co">#&gt; [17] "Richness/Richness_OpenDryPoor.tfw"   "Richness/Richness_OpenDryPoor.tif"  </span></span>
<span><span class="co">#&gt; [19] "Richness/Richness_OpenDryRich.tfw"   "Richness/Richness_OpenDryRich.tif"  </span></span>
<span><span class="co">#&gt; [21] "Richness/Richness_OpenWetPoor.tfw"   "Richness/Richness_OpenWetPoor.tif"  </span></span>
<span><span class="co">#&gt; [23] "Richness/Richness_OpenWetRich.tfw"   "Richness/Richness_OpenWetRich.tif"</span></span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="id_6-references">6 References<a class="anchor" aria-label="anchor" href="#id_6-references"></a>
</h1>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Boyd2022Eco-evolutionary" class="csl-entry">
Boyd, Jennifer Nagel, Jill T. Anderson, Jessica R. Brzyski, Carol J. Baskauf, and J. Cruse-Sanders. 2022. “Eco-Evolutionary Causes and Consequences of Rarity in Plants: A Meta-Analysis.” <em>The New Phytologist</em>. <a href="https://doi.org/10.1111/nph.18172" class="external-link uri">https://doi.org/10.1111/nph.18172</a>.
</div>
<div id="ref-Bregovic2019Contribution" class="csl-entry">
Bregović, Petra, C. Fišer, and M. Zagmajster. 2019. “Contribution of Rare and Common Species to Subterranean Species Richness Patterns.” <em>Ecology and Evolution</em> 9: 11606–18. <a href="https://doi.org/10.1002/ece3.5604" class="external-link uri">https://doi.org/10.1002/ece3.5604</a>.
</div>
<div id="ref-Chapman2018Both" class="csl-entry">
Chapman, Abbie S. A., V. Tunnicliffe, and A. Bates. 2018. “Both Rare and Common Species Make Unique Contributions to Functional Diversity in an Ecosystem Unaffected by Human Activities.” <em>Diversity and Distributions</em> 24: 568–78. <a href="https://doi.org/10.1111/ddi.12712" class="external-link uri">https://doi.org/10.1111/ddi.12712</a>.
</div>
<div id="ref-drake2015range" class="csl-entry">
Drake, John M. 2015. “Range Bagging: A New Method for Ecological Niche Modelling from Presence-Only Data.” <em>Journal of the Royal Society Interface</em> 12 (107): 20150086.
</div>
<div id="ref-guisan2006using" class="csl-entry">
Guisan, Antoine, Olivier Broennimann, Robin Engler, Mathias Vust, Nigel G Yoccoz, Anthony Lehmann, and Niklaus E Zimmermann. 2006. “Using Niche-Based Models to Improve the Sampling of Rare Species.” <em>Conservation Biology</em> 20 (2): 501–11.
</div>
<div id="ref-Jin2019" class="csl-entry">
Jin, Yi, and Hong Qian. 2019. “V.PhyloMaker: An r Package That Can Generate Very Large Phylogenies for Vascular Plants.” <em>Ecography</em> 42: 1353–59.
</div>
<div id="ref-Magurran2003Explaining" class="csl-entry">
Magurran, A., and P. Henderson. 2003. “Explaining the Excess of Rare Species in Natural Species Abundance Distributions.” <em>Nature</em> 422: 714–16. <a href="https://doi.org/10.1038/nature01547" class="external-link uri">https://doi.org/10.1038/nature01547</a>.
</div>
<div id="ref-Proosdij2016Minimum" class="csl-entry">
Proosdij, A. V., M. Sosef, J. Wieringa, and N. Raes. 2016. “Minimum Required Number of Specimen Records to Develop Accurate Species Distribution Models.” <em>Ecography</em> 39: 542–52. <a href="https://doi.org/10.1111/ECOG.01509" class="external-link uri">https://doi.org/10.1111/ECOG.01509</a>.
</div>
<div id="ref-Reddin2015Between-taxon" class="csl-entry">
Reddin, Carl J., J. Bothwell, and J. Lennon. 2015. “Between-Taxon Matching of Common and Rare Species Richness Patterns.” <em>Global Ecology and Biogeography</em> 24: 1476–86. <a href="https://doi.org/10.1111/GEB.12372" class="external-link uri">https://doi.org/10.1111/GEB.12372</a>.
</div>
<div id="ref-Sampaio2023Accurate" class="csl-entry">
Sampaio, A. C. G., and A. Cavalcante. 2023. “Accurate Species Distribution Models: Minimum Required Number of Specimen Records in the Caatinga Biome.” <em>Anais Da Academia Brasileira de Ciencias</em> 95 2: e20201421. <a href="https://doi.org/10.1590/0001-3765202320201421" class="external-link uri">https://doi.org/10.1590/0001-3765202320201421</a>.
</div>
<div id="ref-Saterberg2019A" class="csl-entry">
Säterberg, Torbjörn, T. Jonsson, J. Yearsley, Sofia Berg, and B. Ebenman. 2019. “A Potential Role for Rare Species in Ecosystem Dynamics.” <em>Scientific Reports</em> 9. <a href="https://doi.org/10.1038/s41598-019-47541-6" class="external-link uri">https://doi.org/10.1038/s41598-019-47541-6</a>.
</div>
<div id="ref-Schalkwyk2019Contribution" class="csl-entry">
Schalkwyk, J., J. Pryke, and M. Samways. 2019. “Contribution of Common Vs. Rare Species to Species Diversity Patterns in Conservation Corridors.” <em>Ecological Indicators</em>. <a href="https://doi.org/10.1016/J.ECOLIND.2019.05.014" class="external-link uri">https://doi.org/10.1016/J.ECOLIND.2019.05.014</a>.
</div>
<div id="ref-Stoa2019How" class="csl-entry">
Støa, Bente, R. Halvorsen, J. Stokland, and V. I. Gusarov. 2019. “How Much Is Enough? Influence of Number of Presence Observations on the Performance of Species Distribution Models.” <em>Sommerfeltia</em> 39: 1–28. <a href="https://doi.org/10.2478/som-2019-0001" class="external-link uri">https://doi.org/10.2478/som-2019-0001</a>.
</div>
<div id="ref-Thuiller_2005" class="csl-entry">
Thuiller, Wilfried, Sandra Lavorel, Miguel B. Araújo, Martin T. Sykes, and I. Colin Prentice. 2005. “Climate Change Threats to Plant Diversity in Europe.” <em>Proceedings of the National Academy of Sciences</em> 102 (23): 8245–50. <a href="https://doi.org/10.1073/pnas.0409902102" class="external-link uri">https://doi.org/10.1073/pnas.0409902102</a>.
</div>
<div id="ref-Zanne2014" class="csl-entry">
Zanne, Amy E., David C. Tank, William K. Cornwell, Jonathan M. Eastman, Stephen A. Smith, Richard G. FitzJohn, Daniel J. McGlinn, et al. 2014. “Three Keys to the Radiation of Angiosperms into Freezing Environments.” <em>American Journal of Botany</em> 506: 89–92.
</div>
</div>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/derek-corcoran-barrios/SpeciesPoolR/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/derek-corcoran-barrios/SpeciesPoolR/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing SpeciesPoolR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Derek Corcoran <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-0248-6230" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/derek-corcoran-barrios/SpeciesPoolR/actions/workflows/R-CMD-check.yaml" class="external-link"><img src="https://github.com/derek-corcoran-barrios/SpeciesPoolR/actions/workflows/R-CMD-check.yaml/badge.svg" alt="R-CMD-check"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Derek Corcoran.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
