---
title: "How to use the SpeciesPoolR package"
output:
  bookdown::html_vignette2:
    fig_caption: true
    toc: true
    toc_depth: 2
    number_sections: true
vignette: >
  %\VignetteIndexEntry{How to use the SpeciesPoolR package}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
bibliography: 'https://api.citedrive.com/bib/c81da0f5-66e3-4821-9af9-c556d83cbbd4/references.bib?x=eyJpZCI6ICJjODFkYTBmNS02NmUzLTQ4MjEtOWFmOS1jNTU2ZDgzY2JiZDQiLCAidXNlciI6ICI4NjQ0IiwgInNpZ25hdHVyZSI6ICI3ZDA1NmUzMGRmZTA5YmFmYWIyZTRkNTRhMzljZTViNGY3ZjAwYmIwODgyNWIwOThiMmI2YTYxZDFlYjkxYzVmIn0=.bib'
---



```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# SpeciesPoolR

<!-- badges: start -->
[![R-CMD-check](https://github.com/derek-corcoran-barrios/SpeciesPoolR/actions/workflows/R-CMD-check.yaml/badge.svg)](https://github.com/derek-corcoran-barrios/SpeciesPoolR/actions/workflows/R-CMD-check.yaml)
<!-- badges: end -->

The goal of the SpeciesPoolR package is to generate potential species pools and their summary metrics in a spatial way.
You can install the package directly from GitHub:

```{r, eval=FALSE}
#install.packages("remotes")
remotes::install_github("derek-corcoran-barrios/SpeciesPoolR")
```

No you can load the package

```{r setup}
library(SpeciesPoolR)
```

# Motivation for the pacakge

## Rare species are common and important

In ecological research, the debate on whether rare species outnumber common species within communities is pivotal for understanding biodiversity and guiding conservation efforts.
Numerous studies have shown that rare species typically dominate large ecological assemblages, although common species often exert a more substantial influence on overall species richness patterns [@Magurran2003Explaining; @Bregovic2019Contribution; @Schalkwyk2019Contribution].
This complexity underscores the need for innovative approaches in studying biodiversity, particularly since rare species are challenging to model using traditional Species Distribution Models (SDMs) due to their low occurrence rates [@Boyd2022Eco-evolutionary].

Given the limitations of SDMs in capturing the dynamics of rare species, it is essential to develop alternative methods for integrating these species into biodiversity assessments and conservation planning.
Although rare species contribute uniquely to functional diversity and ecosystem stability, especially in specific habitats [@Chapman2018Both; @Saterberg2019A], their elusiveness in ecological models presents a significant challenge.
The question of the minimum number of presence records required for reliable SDMs is crucial.
Research has shown that while as few as 10-15 presence observations can produce nonrandom models for some species [@Stoa2019How], others require higher thresholdsâ€”ranging from 14 to 25 records depending on the species' prevalence and geographic range [@Proosdij2016Minimum; @Sampaio2023Accurate].
These findings suggest that even sparse datasets can be useful, but the threshold varies significantly depending on species traits and habitat characteristics.
Therefore, researchers must explore novel analytical frameworks and conservation strategies that better accommodate the ecological importance of rare species, thereby enhancing our ability to manage and preserve biodiversity effectively [@Reddin2015Between-taxon].

In highly degraded habitats, such as Denmark, where over 60% of the land is dominated by agriculture and less than 10% remains as natural habitat, traditional SDMs may face further limitations.
The scarcity of natural habitats means that presence records are often skewed towards human-modified landscapes, complicating the modeling of species' ecological preferences.
In such contexts, where the majority of occurrences may not reflect the species' natural behaviors or habitat use, relying on complex SDMs could lead to misleading predictions.
Instead, simpler algorithms that incorporate basic dispersal mechanisms and habitat filtering might be more effective.
By reducing assumptions about habitat preferences, these methods can provide a more realistic framework for conservation planning, particularly when dealing with the restoration of agricultural lands into natural habitats.

For rare species, and indeed for many others, this approach may offer a more practical solution in scenarios where detailed ecological data is sparse or unreliable.
Studies have suggested that in such landscapes, simplistic models that prioritize dispersal and broad habitat suitability over intricate ecological niches can better capture species' potential distributions and their responses to environmental changes [@guisan2006using; @Thuiller_2005], an example to this approach would be range bagging [@drake2015range].
This pragmatic approach is especially pertinent when planning conservation actions in areas where habitat degradation has left little intact nature, and it ensures that even under data constraints, effective biodiversity management can still be pursued.

# Required Data Files

To effectively execute the `SpeciesPoolR` workflow, a set of essential data files must be provided. These files contain the necessary spatial and taxonomic information that underpin the various analytical steps in the package. Below, we detail each required file and its role within the workflow.

## Species List File

- **File Type**: CSV or Excel file

- **Description**: The species list file serves as the foundational dataset, comprising the species of interest for your analysis. At a minimum, this file must include a column for the scientific names of species (`Species`). Additional taxonomic columns, such as `Kingdom`, `Class`, and `Family`, may also be included to facilitate filtering and subgroup analyses.

An example of this file is provided within the package and can be accessed using the following code:

```{r examplespecieslist}
exampleSpecies <- system.file("ex/Species_List.csv", package="SpeciesPoolR")
print(exampleSpecies)
```

This dataset is further discussed in the section on [Reading and Filtering Data](#step-1-reading-and-filtering-data), with a filtered subset displayed in Table \@ref(tab:tablespecies).

## Shapefile 

- **File Type**: Shapefile (.shp)
  
- **Description**: The shapefile delineates the geographic area of interest, which can range from a broad region, such as a country, to a more specific locality, such as a nature reserve. This file is utilized to spatially constrain species occurrences, ensuring that only those within the defined boundaries are included in the analysis.
  
If a shapefile is unavailable, a two-letter country code (e.g., "DK" for Denmark) may be provided as an alternative to specify the area of interest.
  
An example shapefile is included in the package and can be accessed as follows:

```{r exampleshapefile}
shp <- system.file("ex/Aarhus.shp", package="SpeciesPoolR")
print(shp)
```

The shapefile's application is illustrated in the section on [Counting Species Presences](#step-3-counting-species-presences), where it is used to delineate the boundaries of Aarhus commune, as shown in Figure \@ref(fig:plotshapefile).

```{r plotshapefile, echo = F, message=FALSE, error=FALSE, warning=FALSE, fig.cap= "Outline of the comune of Aarhus"}
library(terra)
Aarhus <- terra::vect(shp)
plot(Aarhus)
```


## Raster Template File

- **File Type**: Raster file (e.g., .tif)

- **Description**: The raster template file is employed as a spatial reference for rasterizing species presence buffers. It must cover the entire area of interest and possess a resolution appropriate for the intended analysis. This template ensures consistent spatial alignment across all raster-based operations.

You can explore an example of this file using the following code:

```{r exampletemplate}
template <- system.file("ex/LU_Aarhus.tif", package="SpeciesPoolR")
print(template)
```

The raster template's role in buffer creation is further explained in the section on [Creating Buffers Around Species Presences](#step-1-creating-buffers-around-species-presences), with an example shown in Figure \@ref(fig:plottemplate).

```{r plottemplate, echo = F, message=FALSE, error=FALSE, warning=FALSE, fig.cap= "Raster of the Aarhus comune, the package will use Non NA cells as part of the template"}
library(terra)
Template <- terra::rast(template)
plot(Template)
```


##  Land-Use Raster File

- **File Type**: Raster file (e.g., .tif)
  
- **Description**: This file contains land-use classifications for the study area, where each raster cell is assigned to a specific land-use category (e.g., forest, wetland, urban). This data is crucial for modeling habitat suitability, enabling the filtering of species occurrences based on the prevalent land uses within their potential habitats.
  
An example file is provided in the package:

```{r LUexample}
LU <- system.file("ex/LU_Aarhus.tif", package="SpeciesPoolR")
print(LU)
```

The land-use raster is identical to the template shown in Figure \@ref(fig:plottemplate).

## Land-Use Suitability Raster File

- **File Type**: Raster file (e.g., .tif)
  
- **Description**: This file comprises binary suitability values for various land-use types within the study area, indicating whether each land-use type is suitable (value = 1) or unsuitable (value = 0) for the habitat of interest. The data is subsequently transformed into a long-format table, which is integral to the habitat filtering and species distribution modeling processes.
  
An example raster file is available in the package, and its application is discussed in the section on [Preparing Land-Use Data](#preparing-land-use-data). A visualization of this file is presented in Figure \@ref(fig:plotexampleLU).

```{r plotexampleLU, echo = F, message=FALSE, error=FALSE, warning=FALSE, fig.cap= "Landuse suitability for 8 different landuses in the aarhus commune"}
HabSut <- system.file("ex/HabSut.tif", package = "SpeciesPoolR")
HabSut <- terra::rast(HabSut)
plot(HabSut)
```

# Using SpeciesPoolR Manually

## Importing and Downloading Species Presences

### Step 1: Reading and Filtering Data {#step-1-reading-and-filtering-data}

If you are going to use each of the functions of the SpeciesPoolR manually and sequentially, the first step would be to read in a species list from either a CSV or an XLSX file.
You can use the get_data function for this.
The function allows you to filter your data in a dplyr-like style:

```{r ReadInData}
f <- system.file("ex/Species_List.csv", package="SpeciesPoolR")
filtered_data <- get_data(
   file = f,
   filter = quote(Kingdom == "Plantae" & 
                    Class == "Magnoliopsida" & 
                    Family == "Fabaceae")
)
```

This will generate a dataset that can be used subsequently to count species presences and download species data as seen in table \@ref(tab:tablespecies)

```{r tablespecies, echo = F}
knitr::kable(filtered_data, caption = "Species that will be used to generate species pools")
```

### Step 2: Taxonomic Harmonization

Next, you should perform taxonomic harmonization to ensure that the species names you use are recognized by the GBIF taxonomic backbone.
This can be done using the Clean_Taxa function:

```{r CleanSpecies}
Clean_Species <- SpeciesPoolR::Clean_Taxa(filtered_data$Species)
```

The resulting data frame, with harmonized species names, is shown in table \@ref(tab:cleantable)

```{r cleantable, echo=FALSE}
knitr::kable(Clean_Species, caption = "Taxonomicallty harmonized dataset")
```

### Step 3: Counting Species Presences

After harmonizing the species names, it's important to obtain the number of occurrences of each species in your study area, especially if you plan to calculate rarity.
You can do this using the `count_presences` function.
This function allows you to filter occurrences by country or by a shapefile.
Below is an example for Denmark:

```{r countdenmark}
# Assuming Clean_Species is your data frame
Count_DK <- count_presences(Clean_Species, country = "DK")
```

The resulting data frame of species presences in Denmark is shown in table \@ref(tab:tableCountDenmark)

```{r tableCountDenmark}
knitr::kable(Count_DK, caption = "Counts of presences for the different species within Denmark")
```

Alternatively, you can filter by a specific region using a shapefile.
For example, to count species presences within Aarhus commune:

```{r AarhusCount}
shp <- system.file("ex/Aarhus.shp", package="SpeciesPoolR")

Count_Aarhus <- count_presences(Clean_Species, shapefile = shp)
```

The resulting data.frame for Aarhus commune is shown int table \@ref(tab:tableCountAarhus)

```{r tableCountAarhus, echo = FALSE}
knitr::kable(Count_Aarhus, caption = "Counts of presences for the different species within Aarhus commune")
```

Now it is recommended to eliminate species that have no occurrences in the area, this is done automatically in the workflow version:

```{r filter}
library(data.table)
Count_Aarhus <- Count_Aarhus[N > 0,]
```

So that then we can retrieve the species presences using the function `SpeciesPoolR::get_presences`.

```{r getPresences}
Presences <- get_presences(species = Count_Aarhus$species, shapefile = shp)
```

there we end up with `r nrow(Presences)` presences for our `r nrow(Count_Aarhus)` species.

## Creating Spatial Buffers and Habitat Filtering

### Step 1 Creating Buffers Around Species Presences

Once you have identified the species presences within your area of interest, the next step is to create spatial buffers around these occurrences.
These buffers represent the potential dispersal range of each species, helping to assess areas where the species might establish itself given a specified dispersal distance.

To create these buffers, you'll use a raster file as a template to rasterize the buffers and specify the distance (in meters) representing the species' dispersal range.

```{r buffer500}
Raster <- system.file("ex/LU_Aarhus.tif", package="SpeciesPoolR")

buffer500 <- make_buffer_rasterized(Presences, file = Raster, dist = 500)
```

In this example, the make_buffer_rasterized function generates a 500-meter buffer around each occurrence point in the Presences dataset.
The function utilizes the provided raster file as a template for rasterizing these buffers.

The resulting buffer500 data frame indicates which raster cells are covered by the buffer for each species.
Table \@ref(tab:showbuffer500) displays the first 10 observations of this data frame, providing a detailed view of the bufferâ€™s overlap with raster cells, listing each cell and the corresponding species within that buffer.

```{r showbuffer500, echo=FALSE}
knitr::kable(head(buffer500, 10), caption = "Raster cells within the 500-meter buffer of each species")
```

This table provides a detailed view of how the buffer overlaps with the raster cells, listing each cell and the corresponding species present within that buffer.

### Step 2: Habitat Filtering

After creating the buffers, the next logical step is to filter these areas based on habitat suitability.
This allows you to focus on specific land-use types or habitats where the species is more likely to thrive.
Habitat filtering typically involves using raster data to refine or subset the buffer areas according to the desired habitat criteria.

#### Preparing Land-Use Data

Before you can apply habitat filtering, you need to prepare a long-format land-use table that matches each raster cell to its corresponding habitat types.
This is done using the generate_long_landuse_table function, which takes the path to your raster file and transforms it into a long-format data frame.
The function also filters the data to include only those cells where the suitability value is 1 for at least one land-use type.

```{r long_LU_table}
# Get path for habitat suitability
HabSut <- system.file("ex/HabSut.tif", package = "SpeciesPoolR")

# Generate the long-format land-use table
long_LU_table <- generate_long_landuse_table(path = HabSut)
```

This is crucial for the next steps, the result is shown in table \@ref(tab:longtablehab), as it links each raster cell to potential habitats, enabling you to match species occurrences to suitable environments within their buffer zones.

```{r longtablehab, echo=FALSE}
knitr::kable(head(long_LU_table, 10), caption = "First 10 observations of landuse suitability per cell")
```

#### Applying Habitat Filtering

Once you have the long-format land-use table, you can proceed with habitat filtering.
To achieve this, youâ€™ll use the `ModelAndPredictFunc`, which takes the presence data frame (e.g., Presences) obtained through the get_presences function and the land-use raster.
This comprehensive function encompasses several critical steps:

1- *Grouping Data by Species*: The presence data is grouped by species using `group_split`, ensuring that each species is modeled individually.

2- *Sampling Land-Use Data*: For each species, land-use data is sampled at the presence points using the SampleLanduse function.

3- *Sampling Background Data*: Background points are also sampled from the same land-use raster, providing a contrast to the presence data.

4- *Modeling Habitat Suitability*: The presence and background data are combined and passed to the `ModelSpecies` function.
This function fits a MaxEnt model to predict habitat suitability across the different land-use types.

5- *Predicting Suitability*: The fitted model is then used to predict habitat suitability for each species across all available land-use types.

```{r habitatfiltering}
Habitats <- ModelAndPredictFunc(DF = Presences, file = Raster)
```

The resulting Habitats data frame contains continuous suitability predictions for each species across various land-use types.
Table \@ref(tab:tablespeciespred) shows the first 9 observations, illustrating the predicted habitat suitability scores for the first species in each land-use type.

```{r tablespeciespred}
knitr::kable(Habitats[1:9,], caption = "Predicted habitat suitability scores across various land-use types for the first species. The values represent continuous predictions, indicating the relative likelihood of species presence in each land-use category.")
```

### Step 3: Generating Habitat Suitability Thresholds

While continuous predictions provide a detailed picture of habitat suitability, it is often useful to classify these predictions into binary suitability thresholds.
Thresholds can help determine areas where species presence is more likely or unlikely based on habitat preferences.

The create_thresholds function facilitates this by generating thresholds based on the modeled land-use preferences, using the 90th, 95th, and 99th percentiles of the predicted suitability values.
These thresholds represent the commission rates, helping to define the probability cutoff above which a land-use type is considered suitable for a species.

Hereâ€™s how you can generate these thresholds for the species in your dataset:

```{r thresholds}
Thresholds <- create_thresholds(Model = Habitats, reference = Presences, file = Raster)
```

This will generate de data set with the threshold for the comission rates of 90, 95 and 99th percentile for each species that can be seen in Table \@ref(tab:thresholdtables).

```{r thresholdtables, echo=FALSE}
knitr::kable(Thresholds, digits = 3, caption = "Threshold based on commission rate for the species that are used above")
```

This step produces a data frame containing the thresholds for each species, which can then be used to classify habitat suitability into binary categories, helping you to identify core habitats or areas of higher conservation value.

After we have the continuous thresholds we can generate a lookup table to see which species can inhabit in each landuse type

```{r generateLookup}
LookupTable <- Generate_Lookup(Model = Habitats, Thresholds = Thresholds)
```

This creates Table \@ref(tab:lookuptab), notice how it only shows for each species which habitats are available not the ones that are not.

```{r lookuptab, echo = F}
knitr::kable(LookupTable, caption = "dummy variable that shows which species can inhabit each habitat type")
```

### Step 4: Generating Final Species Presences
  
  In this final step, we apply the `make_final_presences` function to filter the buffered species presences. This filtering process is done in three stages:
  
1. **Lookup Table Filtering:** The function first ensures that each species is only considered in habitats where it can persist based on the species-habitat suitability mappings in the lookup table.

2. **Land-Use Table Filtering:** Next, it filters these suitable habitats to include only those cells where the specific habitat type could exist, based on the long-format land-use table.

3. **Buffer Zone Filtering:** Finally, it restricts the potential species occurrences to areas where the species is likely to disperse, as indicated by the spatial buffers generated around species presence points.

The result is a highly refined dataset that specifies, for each species, the exact cells and habitat types where it can potentially occur, combining habitat suitability, land-use distribution, and species dispersal capability.

```{r createfinalpresences}
final_presences <- make_final_presences(Long_LU_table = long_LU_table, 
                                        Long_Buffer_gbif = buffer500,
                                        LookUpTable = LookupTable)
```

The resulting `final_presences` table provides detailed information on the potential distribution of each species. It specifies which cells and habitats are suitable for each species, ensuring that only the most plausible locations are considered. In table @\ref(tab:finalpresences), you can see the first 15 observations from this final dataset, which represent the potential habitats where each species could thrive, whereas in table @\ref(tab:summaryfinalpresences), you can see a summary of the number of cells that each species could thrive on each habitat type.
  
```{r finalpresences, echo=FALSE}
knitr::kable(head(final_presences, 15), caption = "First 15 rows of the final presences dataset, showing the cells and land-use types where each species can potentially occur")
```


```{r summaryfinalpresences, echo=FALSE}
final_presences |> dplyr::group_by(Landuse, species) |> dplyr::summarise(N = dplyr::n()) |> dplyr::arrange(dplyr::desc(N)) |> knitr::kable(caption = "Summary of number of cells that each species can thrive in for each habitat type")
```


## Generating summary biodiversity statistics

### Step 1 Generating Phylogenetic diversity metrics

In order to generate Phylogenetic Diversity measures, the first step is to generate a phylogenetic tree with the species we have, for that we will use the V.Phylomaker package function `phylo.maker`based on the megaphylogeny of vascular plants [@Jin2019; @Zanne2014], this means that we can only use this functions in species pools of plants.

In this case we use the `generate_tree` from SpeciesPoolR to do so:

```{r treegeneration}
tree <- generate_tree(Count_Aarhus)
```

# Running the SpeciesPoolR Workflow

If you prefer to automate the process and run the `SpeciesPoolR` workflow as a pipeline, you can use the `run_workflow` function.
This function sets up a `targets` workflow that sequentially executes the steps for cleaning species data, counting species presences, and performing spatial analysis.
This approach is especially useful for larger datasets or when you want to ensure reproducibility.

To run the workflow, you can use the following code.
We'll use the same species filter as before, focusing on the `Plantae` kingdom, `Magnoliopsida` class, and `Fabaceae` family.
Additionally, we'll focus on the Aarhus commune using a shapefile.

```{r run_workflow}
shp <- system.file("ex/Aarhus.shp", package = "SpeciesPoolR")
Raster <- system.file("ex/LU_Aarhus.tif", package="SpeciesPoolR")
HabSut <- system.file("ex/HabSut.tif", package = "SpeciesPoolR")


run_workflow(
  file_path = system.file("ex/Species_List.csv", package = "SpeciesPoolR"),
  filter = quote(Kingdom == "Plantae" & Class == "Magnoliopsida" & Family == "Fabaceae"),
  shapefile = shp,
  dist = 500,
  rastertemp = Raster,
  rasterLU = Raster,
  LanduseSuitability = HabSut
)
```

## How It Works

The `run_workflow` function creates a pipeline that:

1. **Reads the data** from the specified file path.

2. **Filters the data** using the provided filter expression.

3. **Cleans the species names** to match the GBIF taxonomic backbone.

4. **Counts the species presences** within the specified geographic area (in this case, Aarhus).

5. **Generates a buffer** around the species presences within the specified distance, using a template raster.

6. **Prepares the land-use data** by generating a long-format table that matches each raster cell to its corresponding habitat types.

7. **Predicts habitat suitability** for each species across different land-use types using the `ModelAndPredictFunc`, which models habitat preferences and provides continuous predictions.

8. **Generates habitat suitability thresholds** for each species based on the predicted suitability scores, using the `create_thresholds` function to define the 90th, 95th, and 99th percentile thresholds.

9. **Builds a lookup table** to determine the land-use types each species can inhabit based on the thresholds.

10. **Generates the final species presences** by filtering the buffered presences according to both the lookup table and the long land-use table, ensuring each species' potential distribution is consistent with its habitat preferences.

11. **Generates a phylogenetic tree** for the species in the species list, using the `generate_tree` function.

12. **Generates a visual representation** of the workflow if `plot = TRUE`.

You can monitor the progress of the workflow and visualize the dependencies between steps using targets::tar_visnetwork().
The result will be similar to running the steps manually but with the added benefits of parallel execution and reproducibility.

This automated approach allows you to streamline your analysis and ensures that all steps are consistently applied to your data.
It also makes it easier to rerun the workflow with different parameters or datasets.

```{r removefiles, include=FALSE}
unlink("_targets/", recursive = T, force = T)
file.remove("_targets.R")
```

# References
