<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>How to use the SpeciesPoolR package • SpeciesPoolR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="How to use the SpeciesPoolR package">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SpeciesPoolR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/how_to_use.html">How to use the SpeciesPoolR package</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/derek-corcoran-barrios/SpeciesPoolR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">



<link href="how_to_use_files/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="how_to_use_files/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="how_to_use_files/vis-9.1.0/vis-network.min.css" rel="stylesheet">
<script src="how_to_use_files/vis-9.1.0/vis-network.min.js"></script><script src="how_to_use_files/visNetwork-binding-2.1.2/visNetwork.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>How to use the SpeciesPoolR package</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/derek-corcoran-barrios/SpeciesPoolR/blob/master/vignettes/how_to_use.Rmd" class="external-link"><code>vignettes/how_to_use.Rmd</code></a></small>
      <div class="d-none name"><code>how_to_use.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="speciespoolr">SpeciesPoolR<a class="anchor" aria-label="anchor" href="#speciespoolr"></a>
</h2>
<!-- badges: start -->
<p><a href="https://github.com/derek-corcoran-barrios/SpeciesPoolR/actions/workflows/R-CMD-check.yaml" class="external-link"><img src="https://github.com/derek-corcoran-barrios/SpeciesPoolR/actions/workflows/R-CMD-check.yaml/badge.svg" alt="R-CMD-check"></a> <!-- badges: end --></p>
<p>The goal of the SpeciesPoolR package is to generate potential species
pools and their summary metrics in a spatial way. You can install the
package directly from GitHub:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#install.packages("remotes")</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"derek-corcoran-barrios/SpeciesPoolR"</span><span class="op">)</span></span></code></pre></div>
<p>No you can load the package</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/derek-corcoran-barrios/SpeciesPoolR" class="external-link">SpeciesPoolR</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="motivation-for-the-pacakge">Motivation for the pacakge<a class="anchor" aria-label="anchor" href="#motivation-for-the-pacakge"></a>
</h2>
<div class="section level3">
<h3 id="rare-species-are-common-and-important">Rare species are common and important<a class="anchor" aria-label="anchor" href="#rare-species-are-common-and-important"></a>
</h3>
<p>In ecological research, the debate on whether rare species outnumber
common species within communities is pivotal for understanding
biodiversity and guiding conservation efforts. Numerous studies have
shown that rare species typically dominate large ecological assemblages,
although common species often exert a more substantial influence on
overall species richness patterns <span class="citation">(Magurran and
Henderson 2003; Bregović, Fišer, and Zagmajster 2019; Schalkwyk, Pryke,
and Samways 2019)</span>. This complexity underscores the need for
innovative approaches in studying biodiversity, particularly since rare
species are challenging to model using traditional Species Distribution
Models (SDMs) due to their low occurrence rates <span class="citation">(Boyd et al. 2022)</span>.</p>
<p>Given the limitations of SDMs in capturing the dynamics of rare
species, it is essential to develop alternative methods for integrating
these species into biodiversity assessments and conservation planning.
Although rare species contribute uniquely to functional diversity and
ecosystem stability, especially in specific habitats <span class="citation">(Chapman, Tunnicliffe, and Bates 2018; Säterberg et al.
2019)</span>, their elusiveness in ecological models presents a
significant challenge. The question of the minimum number of presence
records required for reliable SDMs is crucial. Research has shown that
while as few as 10-15 presence observations can produce nonrandom models
for some species <span class="citation">(Støa et al. 2019)</span>,
others require higher thresholds—ranging from 14 to 25 records depending
on the species’ prevalence and geographic range <span class="citation">(Proosdij et al. 2016; Sampaio and Cavalcante
2023)</span>. These findings suggest that even sparse datasets can be
useful, but the threshold varies significantly depending on species
traits and habitat characteristics. Therefore, researchers must explore
novel analytical frameworks and conservation strategies that better
accommodate the ecological importance of rare species, thereby enhancing
our ability to manage and preserve biodiversity effectively <span class="citation">(Reddin, Bothwell, and Lennon 2015)</span>.</p>
<p>In highly degraded habitats, such as Denmark, where over 60% of the
land is dominated by agriculture and less than 10% remains as natural
habitat, traditional SDMs may face further limitations. The scarcity of
natural habitats means that presence records are often skewed towards
human-modified landscapes, complicating the modeling of species’
ecological preferences. In such contexts, where the majority of
occurrences may not reflect the species’ natural behaviors or habitat
use, relying on complex SDMs could lead to misleading predictions.
Instead, simpler algorithms that incorporate basic dispersal mechanisms
and habitat filtering might be more effective. By reducing assumptions
about habitat preferences, these methods can provide a more realistic
framework for conservation planning, particularly when dealing with the
restoration of agricultural lands into natural habitats.</p>
<p>For rare species, and indeed for many others, this approach may offer
a more practical solution in scenarios where detailed ecological data is
sparse or unreliable. Studies have suggested that in such landscapes,
simplistic models that prioritize dispersal and broad habitat
suitability over intricate ecological niches can better capture species’
potential distributions and their responses to environmental changes
<span class="citation">(GUISAN et al. 2006; Thuiller et al.
2005)</span>, an example to this approach would be range bagging <span class="citation">(Drake 2015)</span>. This pragmatic approach is
especially pertinent when planning conservation actions in areas where
habitat degradation has left little intact nature, and it ensures that
even under data constraints, effective biodiversity management can still
be pursued.</p>
</div>
</div>
<div class="section level2">
<h2 id="using-speciespoolr-manually">Using SpeciesPoolR Manually<a class="anchor" aria-label="anchor" href="#using-speciespoolr-manually"></a>
</h2>
<div class="section level3">
<h3 id="importing-and-downloading-species-presences">Importing and Downloading Species Presences<a class="anchor" aria-label="anchor" href="#importing-and-downloading-species-presences"></a>
</h3>
<div class="section level4">
<h4 id="step-1-reading-and-filtering-data">Step 1: Reading and Filtering Data<a class="anchor" aria-label="anchor" href="#step-1-reading-and-filtering-data"></a>
</h4>
<p>If you are going to use each of the functions of the SpeciesPoolR
manually and sequentially, the first step would be to read in a species
list from either a CSV or an XLSX file. You can use the get_data
function for this. The function allows you to filter your data in a
dplyr-like style:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/Species_List.csv"</span>, package<span class="op">=</span><span class="st">"SpeciesPoolR"</span><span class="op">)</span></span>
<span><span class="va">filtered_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_data.html">get_data</a></span><span class="op">(</span></span>
<span>   file <span class="op">=</span> <span class="va">f</span>,</span>
<span>   filter <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/substitute.html" class="external-link">quote</a></span><span class="op">(</span><span class="va">Kingdom</span> <span class="op">==</span> <span class="st">"Plantae"</span> <span class="op">&amp;</span> </span>
<span>                    <span class="va">Class</span> <span class="op">==</span> <span class="st">"Magnoliopsida"</span> <span class="op">&amp;</span> </span>
<span>                    <span class="va">Family</span> <span class="op">==</span> <span class="st">"Fabaceae"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">200</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">8</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> ","</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">chr</span> (8): redlist_2010, Kingdom, Phyllum, Class, Order, Family, Genus, Species</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span></code></pre></div>
<p>This will generate a dataset that can be used subsequently to count
species presences and download species data as seen in table
@ref(tab:tablespecies)</p>
<table class="table">
<caption>Species that will be used to generate species pools</caption>
<colgroup>
<col width="13%">
<col width="8%">
<col width="14%">
<col width="14%">
<col width="8%">
<col width="9%">
<col width="10%">
<col width="22%">
</colgroup>
<thead><tr class="header">
<th align="left">redlist_2010</th>
<th align="left">Kingdom</th>
<th align="left">Phyllum</th>
<th align="left">Class</th>
<th align="left">Order</th>
<th align="left">Family</th>
<th align="left">Genus</th>
<th align="left">Species</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">NA</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Genista</td>
<td align="left">Genista tinctoria</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Trifolium</td>
<td align="left">Trifolium vesiculosum</td>
</tr>
<tr class="even">
<td align="left">LC</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sativa</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Lathyrus</td>
<td align="left">Lathyrus latifolius</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Anthyllis</td>
<td align="left">Anthyllis vulneraria</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Lathyrus</td>
<td align="left">Lathyrus japonicus</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">Plantae</td>
<td align="left">Magnoliophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia villosa</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="step-2-taxonomic-harmonization">Step 2: Taxonomic Harmonization<a class="anchor" aria-label="anchor" href="#step-2-taxonomic-harmonization"></a>
</h4>
<p>Next, you should perform taxonomic harmonization to ensure that the
species names you use are recognized by the GBIF taxonomic backbone.
This can be done using the Clean_Taxa function:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Clean_Species</span> <span class="op">&lt;-</span> <span class="fu">SpeciesPoolR</span><span class="fu">::</span><span class="fu"><a href="../reference/Clean_Taxa.html">Clean_Taxa</a></span><span class="op">(</span><span class="va">filtered_data</span><span class="op">$</span><span class="va">Species</span><span class="op">)</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(Taxa)`</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(matched_name2)`</span></span></code></pre></div>
<p>The resulting data frame, with harmonized species names, is shown in
table @ref(tab:cleantable)</p>
<table class="table">
<caption>Taxonomicallty harmonized dataset</caption>
<colgroup>
<col width="13%">
<col width="13%">
<col width="6%">
<col width="13%">
<col width="4%">
<col width="7%">
<col width="8%">
<col width="4%">
<col width="5%">
<col width="5%">
<col width="13%">
<col width="4%">
</colgroup>
<thead><tr class="header">
<th align="left">Taxa</th>
<th align="left">matched_name2</th>
<th align="right">confidence</th>
<th align="left">canonicalName</th>
<th align="left">kingdom</th>
<th align="left">phylum</th>
<th align="left">class</th>
<th align="left">order</th>
<th align="left">family</th>
<th align="left">genus</th>
<th align="left">species</th>
<th align="left">rank</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Vicia sepium</td>
<td align="left">Vicia sepium</td>
<td align="right">99</td>
<td align="left">Vicia sepium</td>
<td align="left">Plantae</td>
<td align="left">Tracheophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sepium</td>
<td align="left">SPECIES</td>
</tr>
<tr class="even">
<td align="left">Genista tinctoria</td>
<td align="left">Genista tinctoria</td>
<td align="right">99</td>
<td align="left">Genista tinctoria</td>
<td align="left">Plantae</td>
<td align="left">Tracheophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Genista</td>
<td align="left">Genista tinctoria</td>
<td align="left">SPECIES</td>
</tr>
<tr class="odd">
<td align="left">Trifolium vesiculosum</td>
<td align="left">Trifolium vesiculosum</td>
<td align="right">99</td>
<td align="left">Trifolium vesiculosum</td>
<td align="left">Plantae</td>
<td align="left">Tracheophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Trifolium</td>
<td align="left">Trifolium vesiculosum</td>
<td align="left">SPECIES</td>
</tr>
<tr class="even">
<td align="left">Vicia sativa</td>
<td align="left">Vicia sativa</td>
<td align="right">97</td>
<td align="left">Vicia sativa</td>
<td align="left">Plantae</td>
<td align="left">Tracheophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sativa</td>
<td align="left">SPECIES</td>
</tr>
<tr class="odd">
<td align="left">Lathyrus latifolius</td>
<td align="left">Lathyrus latifolius</td>
<td align="right">98</td>
<td align="left">Lathyrus latifolius</td>
<td align="left">Plantae</td>
<td align="left">Tracheophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Lathyrus</td>
<td align="left">Lathyrus latifolius</td>
<td align="left">SPECIES</td>
</tr>
<tr class="even">
<td align="left">Anthyllis vulneraria</td>
<td align="left">Anthyllis vulneraria</td>
<td align="right">97</td>
<td align="left">Anthyllis vulneraria</td>
<td align="left">Plantae</td>
<td align="left">Tracheophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Anthyllis</td>
<td align="left">Anthyllis vulneraria</td>
<td align="left">SPECIES</td>
</tr>
<tr class="odd">
<td align="left">Lathyrus japonicus</td>
<td align="left">Lathyrus japonicus</td>
<td align="right">99</td>
<td align="left">Lathyrus japonicus</td>
<td align="left">Plantae</td>
<td align="left">Tracheophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Lathyrus</td>
<td align="left">Lathyrus japonicus</td>
<td align="left">SPECIES</td>
</tr>
<tr class="even">
<td align="left">Vicia villosa</td>
<td align="left">Vicia villosa</td>
<td align="right">97</td>
<td align="left">Vicia villosa</td>
<td align="left">Plantae</td>
<td align="left">Tracheophyta</td>
<td align="left">Magnoliopsida</td>
<td align="left">Fabales</td>
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia villosa</td>
<td align="left">SPECIES</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="step-3-counting-species-presences">Step 3: Counting Species Presences<a class="anchor" aria-label="anchor" href="#step-3-counting-species-presences"></a>
</h4>
<p>After harmonizing the species names, it’s important to obtain the
number of occurrences of each species in your study area, especially if
you plan to calculate rarity. You can do this using the
<code>count_presences</code> function. This function allows you to
filter occurrences by country or by a shapefile. Below is an example for
Denmark:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Assuming Clean_Species is your data frame</span></span>
<span><span class="va">Count_DK</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/count_presences.html">count_presences</a></span><span class="op">(</span><span class="va">Clean_Species</span>, country <span class="op">=</span> <span class="st">"DK"</span><span class="op">)</span></span></code></pre></div>
<p>The resulting data frame of species presences in Denmark is shown in
table @ref(tab:tableCountDenmark)</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">Count_DK</span>, caption <span class="op">=</span> <span class="st">"Counts of presences for the different species within Denmark"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Counts of presences for the different species within
Denmark</caption>
<thead><tr class="header">
<th align="left">family</th>
<th align="left">genus</th>
<th align="left">species</th>
<th align="right">N</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sepium</td>
<td align="right">2897</td>
</tr>
<tr class="even">
<td align="left">Fabaceae</td>
<td align="left">Genista</td>
<td align="left">Genista tinctoria</td>
<td align="right">988</td>
</tr>
<tr class="odd">
<td align="left">Fabaceae</td>
<td align="left">Trifolium</td>
<td align="left">Trifolium vesiculosum</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sativa</td>
<td align="right">17379</td>
</tr>
<tr class="odd">
<td align="left">Fabaceae</td>
<td align="left">Lathyrus</td>
<td align="left">Lathyrus latifolius</td>
<td align="right">684</td>
</tr>
<tr class="even">
<td align="left">Fabaceae</td>
<td align="left">Anthyllis</td>
<td align="left">Anthyllis vulneraria</td>
<td align="right">8876</td>
</tr>
<tr class="odd">
<td align="left">Fabaceae</td>
<td align="left">Lathyrus</td>
<td align="left">Lathyrus japonicus</td>
<td align="right">3904</td>
</tr>
<tr class="even">
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia villosa</td>
<td align="right">243</td>
</tr>
</tbody>
</table>
<p>Alternatively, you can filter by a specific region using a shapefile.
For example, to count species presences within Aarhus commune:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">shp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/Aarhus.shp"</span>, package<span class="op">=</span><span class="st">"SpeciesPoolR"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Count_Aarhus</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/count_presences.html">count_presences</a></span><span class="op">(</span><span class="va">Clean_Species</span>, shapefile <span class="op">=</span> <span class="va">shp</span><span class="op">)</span></span></code></pre></div>
<p>The resulting data.frame for Aarhus commune is shown int table
@ref(tab:tableCountAarhus)</p>
<table class="table">
<caption>Counts of presences for the different species within Aarhus
commune</caption>
<thead><tr class="header">
<th align="left">family</th>
<th align="left">genus</th>
<th align="left">species</th>
<th align="right">N</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sepium</td>
<td align="right">283</td>
</tr>
<tr class="even">
<td align="left">Fabaceae</td>
<td align="left">Genista</td>
<td align="left">Genista tinctoria</td>
<td align="right">27</td>
</tr>
<tr class="odd">
<td align="left">Fabaceae</td>
<td align="left">Trifolium</td>
<td align="left">Trifolium vesiculosum</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia sativa</td>
<td align="right">467</td>
</tr>
<tr class="odd">
<td align="left">Fabaceae</td>
<td align="left">Lathyrus</td>
<td align="left">Lathyrus latifolius</td>
<td align="right">41</td>
</tr>
<tr class="even">
<td align="left">Fabaceae</td>
<td align="left">Anthyllis</td>
<td align="left">Anthyllis vulneraria</td>
<td align="right">153</td>
</tr>
<tr class="odd">
<td align="left">Fabaceae</td>
<td align="left">Lathyrus</td>
<td align="left">Lathyrus japonicus</td>
<td align="right">39</td>
</tr>
<tr class="even">
<td align="left">Fabaceae</td>
<td align="left">Vicia</td>
<td align="left">Vicia villosa</td>
<td align="right">10</td>
</tr>
</tbody>
</table>
<p>Now it is recommended to eliminate species that have no occurrences
in the area, this is done automatically in the workflow version:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com" class="external-link">data.table</a></span><span class="op">)</span></span>
<span><span class="va">Count_Aarhus</span> <span class="op">&lt;-</span> <span class="va">Count_Aarhus</span><span class="op">[</span><span class="va">N</span> <span class="op">&gt;</span> <span class="fl">0</span>,<span class="op">]</span></span></code></pre></div>
<p>So that then we can retrieve the species presences using the function
<code><a href="../reference/get_presences.html">SpeciesPoolR::get_presences</a></code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Presences</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_presences.html">get_presences</a></span><span class="op">(</span>species <span class="op">=</span> <span class="va">Count_Aarhus</span><span class="op">$</span><span class="va">species</span>, shapefile <span class="op">=</span> <span class="va">shp</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Geometry created: POLYGON ((10.401438 56.302419, 10.048024 56.355225, 9.886316 56.019928, 10.239729 55.966657, 10.401438 56.302419))"</span></span>
<span><span class="co">#&gt; Starting species 1</span></span>
<span><span class="co">#&gt; 1 of 7 ready! 2024-08-30 04:46:07.835177</span></span>
<span><span class="co">#&gt; Starting species 2</span></span>
<span><span class="co">#&gt; 2 of 7 ready! 2024-08-30 04:46:08.179144</span></span>
<span><span class="co">#&gt; Starting species 3</span></span>
<span><span class="co">#&gt; 3 of 7 ready! 2024-08-30 04:46:08.986255</span></span>
<span><span class="co">#&gt; Starting species 4</span></span>
<span><span class="co">#&gt; 4 of 7 ready! 2024-08-30 04:46:09.344321</span></span>
<span><span class="co">#&gt; Starting species 5</span></span>
<span><span class="co">#&gt; 5 of 7 ready! 2024-08-30 04:46:09.867147</span></span>
<span><span class="co">#&gt; Starting species 6</span></span>
<span><span class="co">#&gt; 6 of 7 ready! 2024-08-30 04:46:10.22466</span></span>
<span><span class="co">#&gt; Starting species 7</span></span>
<span><span class="co">#&gt; 7 of 7 ready! 2024-08-30 04:46:10.56289</span></span></code></pre></div>
<p>there we end up with 1073 presences for our 7 species.</p>
</div>
</div>
<div class="section level3">
<h3 id="creating-spatial-buffers-and-habitat-filtering">Creating Spatial Buffers and Habitat Filtering<a class="anchor" aria-label="anchor" href="#creating-spatial-buffers-and-habitat-filtering"></a>
</h3>
<div class="section level4">
<h4 id="step-1-creating-buffers-around-species-presences">Step 1 Creating Buffers Around Species Presences<a class="anchor" aria-label="anchor" href="#step-1-creating-buffers-around-species-presences"></a>
</h4>
<p>Once you have identified the species presences within your area of
interest, the next step is to create spatial buffers around these
occurrences. These buffers represent the potential dispersal range of
each species, helping to assess areas where the species might establish
itself given a specified dispersal distance.</p>
<p>To create these buffers, you’ll use a raster file as a template to
rasterize the buffers and specify the distance (in meters) representing
the species’ dispersal range.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Raster</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/LU_Aarhus.tif"</span>, package<span class="op">=</span><span class="st">"SpeciesPoolR"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">buffer500</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_buffer_rasterized.html">make_buffer_rasterized</a></span><span class="op">(</span><span class="va">Presences</span>, file <span class="op">=</span> <span class="va">Raster</span>, dist <span class="op">=</span> <span class="fl">500</span><span class="op">)</span></span></code></pre></div>
<p>In this example, the make_buffer_rasterized function generates a
500-meter buffer around each occurrence point in the Presences dataset.
The function utilizes the provided raster file as a template for
rasterizing these buffers.</p>
<p>The resulting buffer500 data frame indicates which raster cells are
covered by the buffer for each species. Table @ref(tab:showbuffer500)
displays the first 10 observations of this data frame, providing a
detailed view of the buffer’s overlap with raster cells, listing each
cell and the corresponding species within that buffer.</p>
<table class="table">
<caption>Raster cells within the 500-meter buffer of each
species</caption>
<thead><tr class="header">
<th align="right">cell</th>
<th align="left">species</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">26</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="even">
<td align="right">27</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="odd">
<td align="right">28</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="even">
<td align="right">29</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="odd">
<td align="right">30</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="even">
<td align="right">161</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="odd">
<td align="right">162</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="even">
<td align="right">163</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="odd">
<td align="right">164</td>
<td align="left">Vicia sepium</td>
</tr>
<tr class="even">
<td align="right">165</td>
<td align="left">Vicia sepium</td>
</tr>
</tbody>
</table>
<p>This table provides a detailed view of how the buffer overlaps with
the raster cells, listing each cell and the corresponding species
present within that buffer.</p>
</div>
<div class="section level4">
<h4 id="step-2-habitat-filtering">Step 2: Habitat Filtering<a class="anchor" aria-label="anchor" href="#step-2-habitat-filtering"></a>
</h4>
<p>After creating the buffers, the next logical step is to filter these
areas based on habitat suitability. This allows you to focus on specific
land-use types or habitats where the species is more likely to thrive.
Habitat filtering typically involves using raster data to refine or
subset the buffer areas according to the desired habitat criteria.</p>
<p>To achieve this, you’ll use the <code>ModelAndPredictFunc</code>,
which takes the presence data frame (e.g., Presences) obtained through
the get_presences function and the land-use raster. This comprehensive
function encompasses several critical steps:</p>
<p>1- <em>Grouping Data by Species</em>: The presence data is grouped by
species using <code>group_split</code>, ensuring that each species is
modeled individually.</p>
<p>2- <em>Sampling Land-Use Data</em>: For each species, land-use data
is sampled at the presence points using the SampleLanduse function.</p>
<p>3- <em>Sampling Background Data</em>: Background points are also
sampled from the same land-use raster, providing a contrast to the
presence data.</p>
<p>4- <em>Modeling Habitat Suitability</em>: The presence and background
data are combined and passed to the <code>ModelSpecies</code> function.
This function fits a MaxEnt model to predict habitat suitability across
the different land-use types.</p>
<p>5- <em>Predicting Suitability</em>: The fitted model is then used to
predict habitat suitability for each species across all available
land-use types.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Habitats</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ModelAndPredictFunc.html">ModelAndPredictFunc</a></span><span class="op">(</span>DF <span class="op">=</span> <span class="va">Presences</span>, file <span class="op">=</span> <span class="va">Raster</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: [spatSample] fewer values returned than requested</span></span>
<span><span class="co">#&gt; Warning: [spatSample] fewer values returned than requested</span></span>
<span><span class="co">#&gt; Warning in lognet(xd, is.sparse, ix, jx, y, weights, offset, alpha, nobs, : one</span></span>
<span><span class="co">#&gt; multinomial or binomial class has fewer than 8 observations; dangerous ground</span></span>
<span><span class="co">#&gt; Warning: [spatSample] requested sample size is larger than the number of cells</span></span>
<span><span class="co">#&gt; Warning: [spatSample] more non-NA cells requested than available</span></span>
<span><span class="co">#&gt; Warning in lognet(xd, is.sparse, ix, jx, y, weights, offset, alpha, nobs, : one</span></span>
<span><span class="co">#&gt; multinomial or binomial class has fewer than 8 observations; dangerous ground</span></span></code></pre></div>
<p>The resulting Habitats data frame contains continuous suitability
predictions for each species across various land-use types. Table
@ref(tab:tablespeciespred) shows the first 9 observations, illustrating
the predicted habitat suitability scores for the first species in each
land-use type.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">Habitats</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">9</span>,<span class="op">]</span>, caption <span class="op">=</span> <span class="st">"Predicted habitat suitability scores across various land-use types for the first species. The values represent continuous predictions, indicating the relative likelihood of species presence in each land-use category."</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Predicted habitat suitability scores across various land-use
types for the first species. The values represent continuous
predictions, indicating the relative likelihood of species presence in
each land-use category.</caption>
<thead><tr class="header">
<th align="left">Landuse</th>
<th align="right">Pred</th>
<th align="left">species</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">OpenDryRich</td>
<td align="right">1.0000000</td>
<td align="left">Anthyllis vulneraria</td>
</tr>
<tr class="even">
<td align="left">OpenDryPoor</td>
<td align="right">1.0000000</td>
<td align="left">Anthyllis vulneraria</td>
</tr>
<tr class="odd">
<td align="left">ForestWetRich</td>
<td align="right">0.6656339</td>
<td align="left">Anthyllis vulneraria</td>
</tr>
<tr class="even">
<td align="left">OpenWetRich</td>
<td align="right">0.6656339</td>
<td align="left">Anthyllis vulneraria</td>
</tr>
<tr class="odd">
<td align="left">OpenWetPoor</td>
<td align="right">0.6656339</td>
<td align="left">Anthyllis vulneraria</td>
</tr>
<tr class="even">
<td align="left">Exclude</td>
<td align="right">0.5122440</td>
<td align="left">Anthyllis vulneraria</td>
</tr>
<tr class="odd">
<td align="left">ForestDryRich</td>
<td align="right">0.3093417</td>
<td align="left">Anthyllis vulneraria</td>
</tr>
<tr class="even">
<td align="left">ForestDryPoor</td>
<td align="right">0.2246560</td>
<td align="left">Anthyllis vulneraria</td>
</tr>
<tr class="odd">
<td align="left">Exclude</td>
<td align="right">0.6335459</td>
<td align="left">Genista tinctoria</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="step-3-generating-habitat-suitability-thresholds">Step 3: Generating Habitat Suitability Thresholds<a class="anchor" aria-label="anchor" href="#step-3-generating-habitat-suitability-thresholds"></a>
</h4>
<p>While continuous predictions provide a detailed picture of habitat
suitability, it is often useful to classify these predictions into
binary suitability thresholds. Thresholds can help determine areas where
species presence is more likely or unlikely based on habitat
preferences.</p>
<p>The create_thresholds function facilitates this by generating
thresholds based on the modeled land-use preferences, using the 90th,
95th, and 99th percentiles of the predicted suitability values. These
thresholds represent the commission rates, helping to define the
probability cutoff above which a land-use type is considered suitable
for a species.</p>
<p>Here’s how you can generate these thresholds for the species in your
dataset:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Thresholds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_thresholds.html">create_thresholds</a></span><span class="op">(</span>Model <span class="op">=</span> <span class="va">Habitats</span>, reference <span class="op">=</span> <span class="va">Presences</span>, file <span class="op">=</span> <span class="va">Raster</span><span class="op">)</span></span></code></pre></div>
<p>This will generate de data set with the threshold for the comission
rates of 90, 95 and 99th percentile for each species that can be seen in
Table @ref(tab:thresholdtables).</p>
<table class="table">
<caption>Threshold based on commission rate for the species that are
used above</caption>
<thead><tr class="header">
<th align="left">species</th>
<th align="right">Thres_99</th>
<th align="right">Thres_95</th>
<th align="right">Thres_90</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Anthyllis vulneraria</td>
<td align="right">0.512</td>
<td align="right">0.512</td>
<td align="right">0.512</td>
</tr>
<tr class="even">
<td align="left">Genista tinctoria</td>
<td align="right">0.634</td>
<td align="right">0.634</td>
<td align="right">0.634</td>
</tr>
<tr class="odd">
<td align="left">Lathyrus japonicus</td>
<td align="right">0.407</td>
<td align="right">0.407</td>
<td align="right">0.407</td>
</tr>
<tr class="even">
<td align="left">Lathyrus latifolius</td>
<td align="right">0.634</td>
<td align="right">0.634</td>
<td align="right">0.634</td>
</tr>
<tr class="odd">
<td align="left">Vicia sativa</td>
<td align="right">0.404</td>
<td align="right">0.404</td>
<td align="right">0.404</td>
</tr>
<tr class="even">
<td align="left">Vicia sepium</td>
<td align="right">0.289</td>
<td align="right">0.289</td>
<td align="right">0.289</td>
</tr>
<tr class="odd">
<td align="left">Vicia villosa</td>
<td align="right">0.633</td>
<td align="right">0.633</td>
<td align="right">0.633</td>
</tr>
</tbody>
</table>
<p>This step produces a data frame containing the thresholds for each
species, which can then be used to classify habitat suitability into
binary categories, helping you to identify core habitats or areas of
higher conservation value.</p>
<p>After we have the continuous thresholds we can generate a lookup
table to see which species can inhabit in each landuse type</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">LookupTable</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Generate_Lookup.html">Generate_Lookup</a></span><span class="op">(</span>Model <span class="op">=</span> <span class="va">Habitats</span>, Thresholds <span class="op">=</span> <span class="va">Thresholds</span><span class="op">)</span></span></code></pre></div>
<p>This creates Table @ref(tab:lookuptab), notice how it only shows for
each species which habitats are available not the ones that are not.</p>
<table class="table">
<caption>dummy variable that shows which species can inhabit each
habitat type</caption>
<thead><tr class="header">
<th align="left">species</th>
<th align="left">Landuse</th>
<th align="right">Pres</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Anthyllis vulneraria</td>
<td align="left">OpenDryRich</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Anthyllis vulneraria</td>
<td align="left">OpenDryPoor</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">Anthyllis vulneraria</td>
<td align="left">ForestWetRich</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Anthyllis vulneraria</td>
<td align="left">OpenWetRich</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">Anthyllis vulneraria</td>
<td align="left">OpenWetPoor</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Lathyrus japonicus</td>
<td align="left">OpenDryPoor</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">Vicia sativa</td>
<td align="left">OpenDryPoor</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Vicia sativa</td>
<td align="left">OpenDryRich</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">Vicia sativa</td>
<td align="left">OpenWetPoor</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Vicia sativa</td>
<td align="left">OpenWetRich</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">Vicia sativa</td>
<td align="left">ForestWetRich</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Vicia sativa</td>
<td align="left">ForestDryRich</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">Vicia sepium</td>
<td align="left">ForestWetRich</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Vicia sepium</td>
<td align="left">ForestDryRich</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">Vicia sepium</td>
<td align="left">OpenDryPoor</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Vicia sepium</td>
<td align="left">OpenWetRich</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">Vicia sepium</td>
<td align="left">OpenWetPoor</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Vicia sepium</td>
<td align="left">OpenDryRich</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level3">
<h3 id="generating-summary-biodiversity-statistics">Generating summary biodiversity statistics<a class="anchor" aria-label="anchor" href="#generating-summary-biodiversity-statistics"></a>
</h3>
<div class="section level4">
<h4 id="step-1-generating-phylogenetic-diversity-metrics">Step 1 Generating Phylogenetic diversity metrics<a class="anchor" aria-label="anchor" href="#step-1-generating-phylogenetic-diversity-metrics"></a>
</h4>
<p>In order to generate Phylogenetic Diversity measures, the first step
is to generate a phylogenetic tree with the species we have, for that we
will use the V.Phylomaker package function <code>phylo.maker</code>based
on the megaphylogeny of vascular plants <span class="citation">(Jin and
Qian 2019; Zanne et al. 2014)</span>, this means that we can only use
this functions in species pools of plants.</p>
<p>In this case we use the <code>generate_tree</code> from SpeciesPoolR
to do so:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_tree.html">generate_tree</a></span><span class="op">(</span><span class="va">Count_Aarhus</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "All species in sp.list are present on tree."</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="running-the-speciespoolr-workflow">Running the SpeciesPoolR Workflow<a class="anchor" aria-label="anchor" href="#running-the-speciespoolr-workflow"></a>
</h2>
<p>If you prefer to automate the process and run the
<code>SpeciesPoolR</code> workflow as a pipeline, you can use the
<code>run_workflow</code> function. This function sets up a
<code>targets</code> workflow that sequentially executes the steps for
cleaning species data, counting species presences, and performing
spatial analysis. This approach is especially useful for larger datasets
or when you want to ensure reproducibility.</p>
<p>To run the workflow, you can use the following code. We’ll use the
same species filter as before, focusing on the <code>Plantae</code>
kingdom, <code>Magnoliopsida</code> class, and <code>Fabaceae</code>
family. Additionally, we’ll focus on the Aarhus commune using a
shapefile.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">shp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/Aarhus.shp"</span>, package <span class="op">=</span> <span class="st">"SpeciesPoolR"</span><span class="op">)</span></span>
<span><span class="va">Raster</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/LU_Aarhus.tif"</span>, package<span class="op">=</span><span class="st">"SpeciesPoolR"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/run_workflow.html">run_workflow</a></span><span class="op">(</span></span>
<span>  file_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/Species_List.csv"</span>, package <span class="op">=</span> <span class="st">"SpeciesPoolR"</span><span class="op">)</span>,</span>
<span>  filter <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/substitute.html" class="external-link">quote</a></span><span class="op">(</span><span class="va">Kingdom</span> <span class="op">==</span> <span class="st">"Plantae"</span> <span class="op">&amp;</span> <span class="va">Class</span> <span class="op">==</span> <span class="st">"Magnoliopsida"</span> <span class="op">&amp;</span> <span class="va">Family</span> <span class="op">==</span> <span class="st">"Fabaceae"</span><span class="op">)</span>,</span>
<span>  shapefile <span class="op">=</span> <span class="va">shp</span>,</span>
<span>  dist <span class="op">=</span> <span class="fl">500</span>,</span>
<span>  rastertemp <span class="op">=</span> <span class="va">Raster</span>,</span>
<span>  rasterLU <span class="op">=</span> <span class="va">Raster</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">▶ dispatched target Raster</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched target Landuses</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed target Raster [3.651 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched target shp</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed target shp [0 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched target file</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed target file [0 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched target data</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed target Landuses [3.678 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed target data [0.161 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched target Clean</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed target Clean [1.032 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch Count_Presences_33538e94b3809372</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch Count_Presences_52d72a5ad405e933</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch Count_Presences_33538e94b3809372 [0.461 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch Count_Presences_e70f77d9439a4770</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch Count_Presences_52d72a5ad405e933 [0.539 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch Count_Presences_dea4ef8633a449a1</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch Count_Presences_e70f77d9439a4770 [0.135 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch Count_Presences_69210fc440d13855</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch Count_Presences_dea4ef8633a449a1 [0.138 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch Count_Presences_a61be030e01ebaf5</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch Count_Presences_69210fc440d13855 [0.117 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch Count_Presences_974105e269324d3e</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch Count_Presences_a61be030e01ebaf5 [0.107 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch Count_Presences_37d1f8d5f74d852c</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch Count_Presences_974105e269324d3e [0.107 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch Count_Presences_37d1f8d5f74d852c [0.108 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed pattern Count_Presences</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched target More_than_zero</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed target More_than_zero [0.001 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch Presences_c112b37cd15959d6</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch Presences_af64bac105a08467</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch Presences_af64bac105a08467 [0.929 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch buffer_0e19b8cb545404d2</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch buffer_0e19b8cb545404d2 [0.056 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch Presences_daf8d6353bc80f0c</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch Presences_c112b37cd15959d6 [1.517 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch buffer_626a53b08dfe709d</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch buffer_626a53b08dfe709d [0.067 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch Presences_310adeccf6b44725</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch Presences_310adeccf6b44725 [0.37 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch buffer_b226446ac3154351</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch Presences_daf8d6353bc80f0c [1.042 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch buffer_edb09c8ec5c9a988</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch buffer_b226446ac3154351 [0.132 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch Presences_e65f4227e8299cc4</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch buffer_edb09c8ec5c9a988 [0.153 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch Presences_d4b9dc68293bd5b2</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch Presences_d4b9dc68293bd5b2 [0.355 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch buffer_cae8301e59fc4e01</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch buffer_cae8301e59fc4e01 [0.021 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch Presences_88937156c1302a12</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch Presences_e65f4227e8299cc4 [0.608 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch buffer_0a8436ee3d4f2644</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch buffer_0a8436ee3d4f2644 [0.023 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched target Phylo_Tree</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch Presences_88937156c1302a12 [0.34 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed pattern Presences</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch buffer_a0190cbfdf5f6f1f</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch buffer_a0190cbfdf5f6f1f [0.019 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed pattern buffer</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch ModelAndPredict_0e19b8cb545404d2</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch ModelAndPredict_0e19b8cb545404d2 [0.662 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch ModelAndPredict_626a53b08dfe709d</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch ModelAndPredict_626a53b08dfe709d [9.539 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch ModelAndPredict_b226446ac3154351</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch ModelAndPredict_b226446ac3154351 [2.824 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch ModelAndPredict_edb09c8ec5c9a988</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed target Phylo_Tree [21.961 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch ModelAndPredict_cae8301e59fc4e01</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch ModelAndPredict_cae8301e59fc4e01 [0.522 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch ModelAndPredict_0a8436ee3d4f2644</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch ModelAndPredict_edb09c8ec5c9a988 [10.624 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched branch ModelAndPredict_a0190cbfdf5f6f1f</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch ModelAndPredict_a0190cbfdf5f6f1f [0.131 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed branch ModelAndPredict_0a8436ee3d4f2644 [6.165 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed pattern ModelAndPredict</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched target Thresholds</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed target Thresholds [0.253 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ dispatched target LookUpTable</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ● completed target LookUpTable [0.004 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; ▶ ended pipeline [39.616 seconds]</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; Warning message:</span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; 3 targets produced warnings. Run targets::tar_meta(fields = warnings, complete_only = TRUE) for the messages. </span></span></span>
<span><span class="co"><span style="color: #BB0000;">#&gt; </span></span></span></code></pre></div>
<div class="visNetwork html-widget html-fill-item" id="htmlwidget-ac96cb3ee4656e2e9ec3" style="width:700px;height:432.632880098888px;"></div>
<script type="application/json" data-for="htmlwidget-ac96cb3ee4656e2e9ec3">{"x":{"nodes":{"name":["buffer","Clean","Count_Presences","data","file","Landuses","LookUpTable","ModelAndPredict","More_than_zero","Phylo_Tree","Presences","Raster","shp","Thresholds"],"type":["pattern","stem","pattern","stem","stem","stem","stem","pattern","stem","stem","pattern","stem","stem","stem"],"description":[null,null,null,null,null,null,null,null,null,null,null,null,null,null],"status":["uptodate","uptodate","uptodate","uptodate","uptodate","uptodate","uptodate","uptodate","uptodate","uptodate","uptodate","uptodate","uptodate","uptodate"],"seconds":[0.471,1.032,1.712,0.161,0,3.678,0.004,30.467,0.001,21.961,5.161,3.651,0,0.253],"bytes":[26989,583,1676,525,16964,3169,323,2201,335,654,17109,3169,25548,311],"branches":[7,null,8,null,null,null,null,7,null,null,7,null,null,null],"label":["buffer","Clean","Count_Presences","data","file","Landuses","LookUpTable","ModelAndPredict","More_than_zero","Phylo_Tree","Presences","Raster","shp","Thresholds"],"color":["#354823","#354823","#354823","#354823","#354823","#354823","#354823","#354823","#354823","#354823","#354823","#354823","#354823","#354823"],"id":["buffer","Clean","Count_Presences","data","file","Landuses","LookUpTable","ModelAndPredict","More_than_zero","Phylo_Tree","Presences","Raster","shp","Thresholds"],"level":[7,3,4,2,1,1,9,7,5,6,6,1,1,8],"shape":["square","dot","square","dot","dot","dot","dot","square","dot","dot","square","dot","dot","dot"]},"edges":{"from":["file","Clean","shp","More_than_zero","shp","Count_Presences","More_than_zero","ModelAndPredict","Thresholds","Presences","Raster","Landuses","ModelAndPredict","Presences","Landuses","Presences","data"],"to":["data","Count_Presences","Count_Presences","Presences","Presences","More_than_zero","Phylo_Tree","LookUpTable","LookUpTable","buffer","buffer","Thresholds","Thresholds","Thresholds","ModelAndPredict","ModelAndPredict","Clean"],"arrows":["to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to"]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot","physics":false},"manipulation":{"enabled":false},"edges":{"smooth":{"type":"cubicBezier","forceDirection":"horizontal"}},"physics":{"stabilization":false},"interaction":{"zoomSpeed":1},"layout":{"hierarchical":{"enabled":true,"direction":"LR"}}},"groups":null,"width":null,"height":null,"idselection":{"enabled":false,"style":"width: 150px; height: 26px","useLabels":true,"main":"Select by id"},"byselection":{"enabled":false,"style":"width: 150px; height: 26px","multiple":false,"hideColor":"rgba(200,200,200,0.5)","highlight":false},"main":{"text":"","style":"font-family:Georgia, Times New Roman, Times, serif;font-weight:bold;font-size:20px;text-align:center;"},"submain":null,"footer":null,"background":"rgba(0, 0, 0, 0)","highlight":{"enabled":true,"hoverNearest":false,"degree":{"from":1,"to":1},"algorithm":"hierarchical","hideColor":"rgba(200,200,200,0.5)","labelOnly":true},"collapse":{"enabled":true,"fit":false,"resetHighlight":true,"clusterOptions":null,"keepCoord":true,"labelSuffix":"(cluster)"},"legend":{"width":0.2,"useGroups":false,"position":"right","ncol":1,"stepX":100,"stepY":100,"zoom":true,"nodes":{"label":["Up to date","Pattern","Stem"],"color":["#354823","#899DA4","#899DA4"],"shape":["dot","square","dot"]},"nodesToDataframe":true},"tooltipStay":300,"tooltipStyle":"position: fixed;visibility:hidden;padding: 5px;white-space: nowrap;font-family: verdana;font-size:14px;font-color:#000000;background-color: #f5f4ed;-moz-border-radius: 3px;-webkit-border-radius: 3px;border-radius: 3px;border: 1px solid #808074;box-shadow: 3px 3px 10px rgba(0, 0, 0, 0.2);"},"evals":[],"jsHooks":[]}</script><div class="section level3">
<h3 id="how-it-works">How It Works<a class="anchor" aria-label="anchor" href="#how-it-works"></a>
</h3>
<p>The run_workflow function creates a pipeline that:</p>
<ol style="list-style-type: decimal">
<li><p>Reads the data from the specified file path.</p></li>
<li><p>Filters the data using the provided filter expression.</p></li>
<li><p>Cleans the species names to match the GBIF taxonomic
backbone.</p></li>
<li><p>Counts the species presences within the specified geographic area
(in this case, Aarhus).</p></li>
<li><p>Generates a buffer around the species presences within the
specified distance, for a template raster.</p></li>
<li><p>Predicts habitat suitability for each species across different
land-use types using the ModelAndPredictFunc, which models habitat
preferences and provides continuous predictions.</p></li>
<li><p>Generates a threshold for each species based on presences and the
models that where build.</p></li>
<li><p>Build a lookup table for each species with the suitable
habitats.</p></li>
<li><p>Generates a phyllogenetic tree for the species in the species
list.</p></li>
<li><p>Generates a visual representation of the workflow (if plot =
TRUE).</p></li>
</ol>
<p>You can monitor the progress of the workflow and visualize the
dependencies between steps using targets::tar_visnetwork(). The result
will be similar to running the steps manually but with the added
benefits of parallel execution and reproducibility.</p>
<p>This automated approach allows you to streamline your analysis and
ensures that all steps are consistently applied to your data. It also
makes it easier to rerun the workflow with different parameters or
datasets.</p>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Boyd2022Eco-evolutionary" class="csl-entry">
Boyd, Jennifer Nagel, Jill T. Anderson, Jessica R. Brzyski, Carol J.
Baskauf, and J. Cruse-Sanders. 2022. <span>“Eco-Evolutionary Causes and
Consequences of Rarity in Plants: A Meta-Analysis.”</span> <em>The New
Phytologist</em>. <a href="https://doi.org/10.1111/nph.18172" class="external-link">https://doi.org/10.1111/nph.18172</a>.
</div>
<div id="ref-Bregovic2019Contribution" class="csl-entry">
Bregović, Petra, C. Fišer, and M. Zagmajster. 2019. <span>“Contribution
of Rare and Common Species to Subterranean Species Richness
Patterns.”</span> <em>Ecology and Evolution</em> 9: 11606–18. <a href="https://doi.org/10.1002/ece3.5604" class="external-link">https://doi.org/10.1002/ece3.5604</a>.
</div>
<div id="ref-Chapman2018Both" class="csl-entry">
Chapman, Abbie S. A., V. Tunnicliffe, and A. Bates. 2018. <span>“Both
Rare and Common Species Make Unique Contributions to Functional
Diversity in an Ecosystem Unaffected by Human Activities.”</span>
<em>Diversity and Distributions</em> 24: 568–78. <a href="https://doi.org/10.1111/ddi.12712" class="external-link">https://doi.org/10.1111/ddi.12712</a>.
</div>
<div id="ref-drake2015range" class="csl-entry">
Drake, John M. 2015. <span>“Range Bagging: A New Method for Ecological
Niche Modelling from Presence-Only Data.”</span> <em>Journal of the
Royal Society Interface</em> 12 (107): 20150086.
</div>
<div id="ref-GUISAN_2006" class="csl-entry">
GUISAN, ANTOINE, OLIVIER BROENNIMANN, ROBIN ENGLER, MATHIAS VUST, NIGEL
G. YOCCOZ, ANTHONY LEHMANN, and NIKLAUS E. ZIMMERMANN. 2006.
<span>“Using Niche‐based Models to Improve the Sampling of Rare
Species.”</span> <em>Conservation Biology</em> 20 (2): 501–11. <a href="https://doi.org/10.1111/j.1523-1739.2006.00354.x" class="external-link">https://doi.org/10.1111/j.1523-1739.2006.00354.x</a>.
</div>
<div id="ref-Jin2019" class="csl-entry">
Jin, Yi, and Hong Qian. 2019. <span>“V.PhyloMaker: An r Package That Can
Generate Very Large Phylogenies for Vascular Plants.”</span>
<em>Ecography</em> 42: 1353–59.
</div>
<div id="ref-Magurran2003Explaining" class="csl-entry">
Magurran, A., and P. Henderson. 2003. <span>“Explaining the Excess of
Rare Species in Natural Species Abundance Distributions.”</span>
<em>Nature</em> 422: 714–16. <a href="https://doi.org/10.1038/nature01547" class="external-link">https://doi.org/10.1038/nature01547</a>.
</div>
<div id="ref-Proosdij2016Minimum" class="csl-entry">
Proosdij, A. V., M. Sosef, J. Wieringa, and N. Raes. 2016.
<span>“Minimum Required Number of Specimen Records to Develop Accurate
Species Distribution Models.”</span> <em>Ecography</em> 39: 542–52. <a href="https://doi.org/10.1111/ECOG.01509" class="external-link">https://doi.org/10.1111/ECOG.01509</a>.
</div>
<div id="ref-Reddin2015Between-taxon" class="csl-entry">
Reddin, Carl J., J. Bothwell, and J. Lennon. 2015. <span>“Between-Taxon
Matching of Common and Rare Species Richness Patterns.”</span>
<em>Global Ecology and Biogeography</em> 24: 1476–86. <a href="https://doi.org/10.1111/GEB.12372" class="external-link">https://doi.org/10.1111/GEB.12372</a>.
</div>
<div id="ref-Sampaio2023Accurate" class="csl-entry">
Sampaio, A. C. G., and A. Cavalcante. 2023. <span>“Accurate Species
Distribution Models: Minimum Required Number of Specimen Records in the
Caatinga Biome.”</span> <em>Anais Da Academia Brasileira de
Ciencias</em> 95 2: e20201421. <a href="https://doi.org/10.1590/0001-3765202320201421" class="external-link">https://doi.org/10.1590/0001-3765202320201421</a>.
</div>
<div id="ref-Saterberg2019A" class="csl-entry">
Säterberg, Torbjörn, T. Jonsson, J. Yearsley, Sofia Berg, and B.
Ebenman. 2019. <span>“A Potential Role for Rare Species in Ecosystem
Dynamics.”</span> <em>Scientific Reports</em> 9. <a href="https://doi.org/10.1038/s41598-019-47541-6" class="external-link">https://doi.org/10.1038/s41598-019-47541-6</a>.
</div>
<div id="ref-Schalkwyk2019Contribution" class="csl-entry">
Schalkwyk, J., J. Pryke, and M. Samways. 2019. <span>“Contribution of
Common Vs. Rare Species to Species Diversity Patterns in Conservation
Corridors.”</span> <em>Ecological Indicators</em>. <a href="https://doi.org/10.1016/J.ECOLIND.2019.05.014" class="external-link">https://doi.org/10.1016/J.ECOLIND.2019.05.014</a>.
</div>
<div id="ref-Stoa2019How" class="csl-entry">
Støa, Bente, R. Halvorsen, J. Stokland, and V. I. Gusarov. 2019.
<span>“How Much Is Enough? Influence of Number of Presence Observations
on the Performance of Species Distribution Models.”</span>
<em>Sommerfeltia</em> 39: 1–28. <a href="https://doi.org/10.2478/som-2019-0001" class="external-link">https://doi.org/10.2478/som-2019-0001</a>.
</div>
<div id="ref-Thuiller_2005" class="csl-entry">
Thuiller, Wilfried, Sandra Lavorel, Miguel B. Araújo, Martin T. Sykes,
and I. Colin Prentice. 2005. <span>“Climate Change Threats to Plant
Diversity in Europe.”</span> <em>Proceedings of the National Academy of
Sciences</em> 102 (23): 8245–50. <a href="https://doi.org/10.1073/pnas.0409902102" class="external-link">https://doi.org/10.1073/pnas.0409902102</a>.
</div>
<div id="ref-Zanne2014" class="csl-entry">
Zanne, Amy E., David C. Tank, William K. Cornwell, Jonathan M. Eastman,
Stephen A. Smith, Richard G. FitzJohn, Daniel J. McGlinn, et al. 2014.
<span>“Three Keys to the Radiation of Angiosperms into Freezing
Environments.”</span> <em>American Journal of Botany</em> 506: 89–92.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Derek Corcoran.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
